---
title: "Project: Identification of a Notch transcriptomic signature for breast cancer"
subtitle: "Part 9 - NOTCH network analysis"
author: "Computational Oncology: Felix Geist"
date: "`r Sys.Date()`"
output: 
  html_document:
    theme: flatly
    highlight: tango
    fig_caption: TRUE
    number_sections: FALSE
    code_folding: hide
    toc: TRUE
    toc_depth: 1
    toc_float:
      collapsed: FALSE
      smooth_scroll: FALSE
bibliography: "`r here::here('src','part9.bib')`"
link-citations: true
biblio-style: "apalike"
---

```{r 00_setup, include=FALSE}

knitr::opts_chunk$set(
	fig.height = 4,                  # 4 inch high, 6 inch wide
	fig.width = 6,
	dpi = 300,                       
	dev = c("svg"),                  
	fig.align = "center",
	# fig.path = "../Figures/Part9/",           
	message = FALSE,
	warning = FALSE,
	echo = TRUE,                     
	cache = TRUE                    
)
```

```{css, echo=FALSE}

text {
  font-family: sans-serif;
}

h1, .h1, h2, .h2, h3, .h3 {
    margin-top: 0.5em;
}

div.box { 
  background-color:#EEEEEE; 
  border: 2px solid #0F69AF;
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

div.obs { 
  background-color:white; 
  border: 2px solid #0F69AF; 
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

body, td {
   font-size: 14px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 14px
}

img {
    max-width: 80%;
}
```

```{r 01-packages, echo=TRUE, message=FALSE, warning=FALSE, results='hide'}

library(tidyverse)

library(readxl)
library(xlsx)
library(knitr)
library(kableExtra)

library(ggplot2)
library(ggpubr)
library(ggrepel)

library(circlize)
library(ggthemes)
#library(ComplexHeatmap)
library(rmarkdown)
library(DT)

library(viridis)

### Network Analysis

library(igraph)
library(STRINGdb)
library(msigdbr)

# parallelize the analysis:

library(foreach)
library(doParallel)

# Citations:
knitr::write_bib(file = '../src/part9.bib')

# color schema

merck_colors <- ggthemes::tableau_color_pal("Color Blind")(10)

merck_colors_ext <- colorRamp2(seq(from = 1, to = 40, by = 4), merck_colors[1:10])

subtype_colors <- merck_colors[c(5,9,6)]
names(subtype_colors) <- c("basal", "her2", "luma")

all_subtype_colors <- c(subtype_colors, c("lumb" = as.character(merck_colors[4])))

time_colors <- merck_colors[c(10,7)]
names(time_colors) <- c("8h", "72h")

treatment_colors <- merck_colors[c(1,2,3,4)]
names(treatment_colors) <- c("Notch on", "Notch off", "ground state","Notch on/off")

batch_colors <- merck_colors[c(5,9)]
names(batch_colors) <- c("Validation", "Training")

col_heatmap <- c(merck_colors[1], "white", merck_colors[6])
col_heatmap_not0 <- c(merck_colors[1], merck_colors[3], merck_colors[6])

theme_set(theme_bw())

# heatmap colors:

options(knitr.kable.NA = '')   # This replaces NA in knitr tables to an empty string

```

**Research Questions**

<div class = "box">

We want to identify general applicable signatures for the activation and inactivation of the NOTCH pathway. Therefore, we treated triple negative breast cancer cell lines either with the gamma-secretase inhibitor DAPT or cultivated the cells on immobilized NOTCH ligand Jagged1. Subsequently, the cells were harvested after 8h or 72h of incubation time and their transcriptome analyzed by RNASeq.  

Here, we aim to determine how "central" or close each gene signature is to the Notch pathway by examining their interactions within a larger protein-protein interaction network.
</div>
  
**Methods**

We make use of the STRING database package in R to access the protein-protein interaction networks, selecting the most recent version "12" for humans.
  
As our central reference, we select the NOTCH genes from the KEGG database, which are sourced from the msigdbr package.  
Leveraging the STRING database, genes from both the Notch pathway and our signature lists are mapped to unique STRING IDs. These IDs are critical for querying protein-protein interactions within the STRING database.  
We then retrieve interactions, essentially the relationships, between these mapped genes from the STRING database. This step is crucial to discern which genes have established interactions.
Utilizing these interactions, we construct an interaction network with the assistance of the igraph package. This intricate network aids us in computing the distances between various genes.
Subsequently, we determine the minimal distance between a specific gene and any gene within the Notch pathway. In scenarios where a gene is isolated (meaning it's not integrated into the network) or when the calculated distance is "Infinity", we designate the distance as "NA".  
Following this, we proceed to calculate the mean distance for each gene present in a signature relative to the genes within the Notch pathway.  
We applied this procedure to all gene signatures both our Notch 20 signature and the other published signatures.    

**Conclusion**  

<div class = "box">
In essence, our endeavor is to quantify how "central" or proximal each gene signature is to the Notch pathway, achieved by scrutinizing their interactions within an expansive protein-protein interaction network.
</div>

# Methods and Data Sources

## Methods

**Platform and Report**  

The analysis was performed using R `r paste(R.Version()[c("major", "minor")], collapse = ".")` [@R-base] with the extension of the `tidyverse` [@R-tidyverse; @tidyverse2019] (dplyr, forcats, ggplot2, purrr, readr, stringr, tibble, tidyr).   
The report was generated using `Rmarkdown` [@R-rmarkdown; @rmarkdown2018; @rmarkdown2020] and `knitr` [@R-knitr; @knitr2014; @knitr2015].  

**Network Analysis**  
The KEGG NOTCH signature was revived from the `msigdbr` [@R-msigdbr] and the gene-interaction network from `STRINGdb` [@STRINGdb2021]. The network was subsequently analyzed by using the `igraph` implementation [@R-igraph; @igraph2006].  

**Visualization**  
Plots are drawn by `ggplot2`.  
Tables are drawn, using the `knitr` package, together with `kableExtra` styling [@R-kableExtra]. Interactive Tables are integrated using the `DT` package [@R-DT].  

# Network Analysis

## KEGG Pathway as "central point"

```{r eval=FALSE, fig.height=9, fig.width=12, include=TRUE}
base::load("../RData/Part3.RData")

# signatures to show:

focus_signatures <- c("Notch 20-Gene Signature", sort(c( 
"Braune_2016_TNBC_Notch_sig_CSL_KO",
"Castel_2013_C2C12_Dll1_vs_DAPT_targets_24h_up",
"WP_NOTCH_SIGNALING_PATHWAY",
"GOBP_NOTCH_SIGNALING_INVOLVED_IN_HEART_DEVELOPMENT",
"Labban_2018_SCC_RBPJ_targets",
"Leontovich_2018_NOTCH3_nuclear_reprogramming_BRCA",
"PID_NOTCH_PATHWAY",
"Stoeck2014_TNBC_ACC_GSI_response_1",
"WP_CANONICAL_AND_NONCANONICAL_NOTCH_SIGNALING",
"KEGG_NOTCH_SIGNALING_PATHWAY",
"Poulsen_2018_N1_repressed",
"HALLMARK_NOTCH_SIGNALING",
"BIOCARTA_NOTCH_PATHWAY",
"VILIMAS_NOTCH1_TARGETS_UP",
"GOBP_POSITIVE_REGULATION_OF_NOTCH_SIGNALING_PATHWAY",
"Xu_2017_MDAMB231_RBPJ_targets")))

rm(cs.gs.opt_notch,
   cs.gs.published_notch,
   Training_gsea_summary)

# Load Gene Signatures

gs.list.opt_notch <- lapply(gs.list.opt_notch, unique)
gs.list.published_notch <- lapply(gs.list.published_notch, unique)

all_genes <- data.frame(human_gene_symbol = unique(c(unlist(gs.list.opt_notch),
                                                     unlist(gs.list.published_notch))))

# Initialize the STRING database (use the latest version)
string_db <- STRINGdb::STRINGdb$new(version="12", species=9606) # 9606 is for Homo sapiens

# Fetch Notch signaling-related genes (from KEGG)

MSIGDB_Notch <- msigdbr::msigdbr(species = "Homo sapiens") %>%
  dplyr::filter(gs_name == "KEGG_NOTCH_SIGNALING_PATHWAY")

notch_genes <- MSIGDB_Notch %>% 
  dplyr::select(human_gene_symbol)

# map the genes:
notch_genes_mapped <- string_db$map( as.data.frame(notch_genes), 
                                     "human_gene_symbol", removeUnmappedRows = TRUE ) %>% 
  dplyr::distinct()

cls_gene_mapped <- notch_genes_mapped %>% 
  dplyr::filter(human_gene_symbol == "RBPJ")

all_genes_mapped <- string_db$map( bind_rows(all_genes, notch_genes), 
                                   "human_gene_symbol", removeUnmappedRows = TRUE ) %>% 
  dplyr::distinct()

# # Fetch interactions
# initial_interactions <- string_db$get_interactions(all_genes_mapped$STRING_id)
# 
# # Create the igraph object from combined interactions
# interaction_network <- igraph::graph_from_data_frame(initial_interactions, directed=TRUE)
# 
# save.image("Part9_interaction_image.RData")

######## add neighboring interactions #####

# 1. Fetching Neighboring Genes:
all_genes_neighbors <- string_db$get_neighbors(all_genes_mapped$STRING_id)

# 2. Updating Gene List:
# Unlist and unique to remove duplicates and format as a vector
neighbors <- unique(all_genes_neighbors)
extended_genes <- unique(c(all_genes_mapped$STRING_id, neighbors))

# 3. Fetch interactions for the extended gene list
extended_interactions <- string_db$get_interactions(extended_genes)

# 4. Creating Extended Interaction Network:
# Create a new igraph object with the extended interactions
extended_interaction_network <- igraph::graph_from_data_frame(extended_interactions, directed=TRUE)

# Now you have an extended interaction network and can proceed with your centrality analysis as before

save.image("Part9_extended_interaction_image.RData")
```


```{r fig.height=10, fig.width=15}

base::load("../RData/Part9_extended_interaction_image.RData")

### calculate all pair distances ###

all_pair_distances <- igraph::distances(extended_interaction_network, mode = "all")

####################################

min_distance <- function(STRING_id, distance_to){

  if(is.na(STRING_id) || !STRING_id %in% igraph::V(extended_interaction_network)$name) return(NA)
  
  # Find shortest path to any gene in the central NOTCH pathway
  # shortest_paths <- igraph::distances(extended_interaction_network,
  #                               v = STRING_id,
  #                               to = distance_to,
  #                               mode = "all")
  # 
  # min_distance <- min(shortest_paths, na.rm = TRUE)

  # Index to access precomputed distances
  v_index <- match(STRING_id, igraph::V(extended_interaction_network)$name)
  to_index <- match(distance_to, igraph::V(extended_interaction_network)$name)

  shortest_paths <- all_pair_distances[v_index, to_index]
  min_distance <- min(shortest_paths, na.rm = TRUE)
  
  if(min_distance == Inf){
    return(NA)
  }
  
  return(min_distance)
}

centrality_score <- function(signature_genes, distance_to){

  signatures_distance <- data.frame(human_gene_symbol = signature_genes)
  
  signatures_distance %>% 
      dplyr::left_join(all_genes_mapped %>% 
                         dplyr::filter(human_gene_symbol %in% !!signature_genes) %>% 
                         dplyr::select(human_gene_symbol, STRING_id) %>% 
                         dplyr::distinct(),
                       by = join_by(human_gene_symbol)) %>% 
      dplyr::rowwise() %>% 
      dplyr::mutate(distance = min_distance(STRING_id, distance_to))
}

distance_to_notch_list <- lapply(c(gs.list.opt_notch,gs.list.published_notch), 
                                 function(x) centrality_score(x, notch_genes_mapped$STRING_id))

centrality_score_kegg <- t(bind_rows(lapply(distance_to_notch_list, function(x) mean(x$distance, na.rm = TRUE)))) %>% 
  as.data.frame() %>% 
  rownames_to_column("id_geneset_id")

colnames(centrality_score_kegg) <- c("geneset_id", "centrality score")

centrality_score_kegg <- centrality_score_kegg %>% 
  dplyr::left_join(data.frame(geneset_length = unlist(lapply(c(gs.list.opt_notch,gs.list.published_notch), length))) %>% 
                     rownames_to_column("geneset_id"),
                   by = join_by(geneset_id))

#### sample random background distribution for KEGG as central point

range_signature_lengths <- range(lapply(c(gs.list.opt_notch,gs.list.published_notch), length))

set.seed(1234)

#### 1000 random chosen genes per signature size + parallelize it

random_gene_distance <- function(signature_length, distance_to){
  string_ids <- base::sample(extended_genes, signature_length, replace = FALSE)
  mean(sapply(string_ids, function(x) min_distance(x, distance_to)), na.rm = TRUE)
}

iterations <- 1000                               # used for the foreach loop  

cat("iterations",file="iterations_kegg_centrality.txt",sep="\n",append=FALSE)

cl <- parallel::makeCluster(16)
doParallel::registerDoParallel(cl)

x <- foreach(i=range_signature_lengths[1]:range_signature_lengths[2], .combine='cbind') %dopar% {
  
  cat(i,file="iterations_kegg_centrality.txt",sep="\n",append=TRUE)
  sapply(1:iterations, function(x) random_gene_distance(i, notch_genes_mapped$STRING_id))
  
}

parallel::stopCluster(cl)

colnames(x) <- paste0("Signature_size_", range_signature_lengths[1]:range_signature_lengths[2])

### calculate p-value

get_emp_pvalue <- function(distribution, mean_distance, signature_size){
  sig_length <- distribution[,paste0("Signature_size_", signature_size)]
  sig_smaller <- sig_length[sig_length < mean_distance]
  
  if(length(sig_smaller)==0){
    return(0)
  } else {
    emp_pval <- length(sig_smaller)/length(sig_length)
  }
  
  return(emp_pval)
}

mean_background_distance <- data.frame(geneset_length = as.numeric(str_remove(colnames(x), "Signature_size_")),
                                       mean_background_distance = colMeans(x))

centrality_score_kegg <- centrality_score_kegg %>% 
  dplyr::left_join(mean_background_distance,
                   by = join_by(geneset_length)) %>% 
  dplyr::rowwise() %>% 
  dplyr::mutate(emp_pval = get_emp_pvalue(distribution = !!x, 
                                          mean_distance = `centrality score`, 
                                          signature_size = geneset_length))


##################


xlsx::write.xlsx(file = "centrality_score.xlsx",
                 x = centrality_score_kegg,
                 sheet = "Centrality Score")

xlsx::write.xlsx(file = "centrality_score.xlsx",
                 sheet = "single distances",
                 x = bind_rows(distance_to_notch_list, .id = "id_geneset_id"),
                 append = TRUE)

centrality_score_kegg <- centrality_score_kegg %>%
  dplyr::rename(ID = "geneset_id") %>%
  dplyr::mutate(ID = str_replace_all(ID, "\\_", " ")) %>%
  dplyr::mutate(ID = factor(ID, levels = .$ID[order(.$`centrality score`, decreasing = TRUE)]),
                textlabel = round(`centrality score`, digits = 2)) %>%
  dplyr::mutate(label_columns = ifelse(as.character(ID) == "Notch 20-Gene Signature", "bold", "plain"))

centrality_score_kegg %>% 
  ggplot(aes(x = ID,
             y = `centrality score`,
             color = `centrality score`,
             label = textlabel)) +
  geom_hline(yintercept = mean(centrality_score_kegg$mean_background_distance),
             linetype = "dashed",
             color = "darkgrey") +
  geom_segment(aes(x = ID,
                   xend = ID,
                   y = 0,
                   yend = `centrality score`),
               color = "darkgrey",
               linewidth = 1) +
  geom_point(size = 6,
             shape = 16) +
  geom_text(color = "white",
            fontface = "bold",
            size = 2) + 
  theme(axis.text.x = element_text(face = centrality_score_kegg$label_columns[order(centrality_score_kegg$`centrality score`, decreasing = TRUE)])) +
  scale_color_gradient(low = merck_colors[1],
                      high = merck_colors[6],
                      limits = c(0,max(centrality_score_kegg$`centrality score`)),
                      na.value = merck_colors[1]) +
  rotate_x_text(angle = 90) +
  coord_cartesian(ylim = c(0, max(centrality_score_kegg$`centrality score`)*1.1)) +
  labs(title = "Mean Centrality Score")


```

## CSL as "central point"

```{r fig.height=10, fig.width=15}

distance_to_csl_list <- lapply(c(gs.list.opt_notch,gs.list.published_notch), function(x) centrality_score(x, cls_gene_mapped$STRING_id))

centrality_score_csl <- t(bind_rows(lapply(distance_to_csl_list, function(x) mean(x$distance, na.rm = TRUE)))) %>% 
  as.data.frame() %>% 
  rownames_to_column("id_geneset_id")

colnames(centrality_score_csl) <- c("geneset_id", "centrality score")

centrality_score_csl <- centrality_score_csl %>% 
  dplyr::left_join(data.frame(geneset_length = unlist(lapply(c(gs.list.opt_notch,gs.list.published_notch), length))) %>% 
                     rownames_to_column("geneset_id"),
                   by = join_by(geneset_id))

#### sample random background distribution for KEGG as central point

set.seed(1234)
cat("iterations",file="iterations_csl_centrality.txt",sep="\n",append=FALSE)

#### 1000 random chosen genes per signature size + parallelize it

cl <- parallel::makeCluster(20)
doParallel::registerDoParallel(cl)

x <- foreach(i=range_signature_lengths[1]:range_signature_lengths[2], .combine='cbind') %dopar% {
  
  cat("iterations",file="iterations_csl_centrality.txt",sep="\n",append=TRUE)
  sapply(1:iterations, function(x) random_gene_distance(i, cls_gene_mapped$STRING_id))
  
}

parallel::stopCluster(cl)

colnames(x) <- paste0("Signature_size_", range_signature_lengths[1]:range_signature_lengths[2])

mean_background_distance <- data.frame(geneset_length = as.numeric(str_remove(colnames(x), "Signature_size_")),
                                       mean_background_distance = colMeans(x))

centrality_score_csl <- centrality_score_csl %>% 
  dplyr::left_join(mean_background_distance,
                   by = join_by(geneset_length)) %>% 
  dplyr::rowwise() %>% 
  dplyr::mutate(emp_pval = get_emp_pvalue(distribution = !!x, 
                                          mean_distance = `centrality score`, 
                                          signature_size = geneset_length))


##################


xlsx::write.xlsx(file = "centrality_score_csl.xlsx",
                 x = centrality_score_csl,
                 sheet = "Centrality Score")

xlsx::write.xlsx(file = "centrality_score_csl.xlsx",
                 sheet = "single distances",
                 x = bind_rows(distance_to_csl_list, .id = "id_geneset_id"),
                 append = TRUE)

centrality_score_csl <- centrality_score_csl %>%
  dplyr::rename(ID = "geneset_id") %>%
  dplyr::mutate(ID = str_replace_all(ID, "\\_", " ")) %>%
  dplyr::mutate(ID = factor(ID, levels = .$ID[order(.$`centrality score`, decreasing = TRUE)]),
                textlabel = round(`centrality score`, digits = 2)) %>%
  dplyr::mutate(label_columns = ifelse(as.character(ID) == "Notch 20-Gene Signature", "bold", "plain"))

centrality_score_csl %>% 
  ggplot(aes(x = ID,
             y = `centrality score`,
             color = `centrality score`,
             label = textlabel)) +
  geom_hline(yintercept = mean(centrality_score_csl$mean_background_distance),
             linetype = "dashed",
             color = "darkgrey") +
  geom_segment(aes(x = ID,
                   xend = ID,
                   y = 0,
                   yend = `centrality score`),
               color = "darkgrey",
               linewidth = 1) +
  geom_point(size = 6,
             shape = 16) +
  geom_text(color = "white",
            fontface = "bold",
            size = 2) + 
  theme(axis.text.x = element_text(face = centrality_score_csl$label_columns[order(centrality_score_csl$`centrality score`, decreasing = TRUE)])) +
  scale_color_gradient(low = merck_colors[1],
                      high = merck_colors[6],
                      limits = c(0,max(centrality_score_csl$`centrality score`)),
                      na.value = merck_colors[1]) +
  rotate_x_text(angle = 90) +
  coord_cartesian(ylim = c(0, max(centrality_score_csl$`centrality score`)*1.1)) +
  labs(title = "Mean Centrality Score")


```


# References {.unlisted .unnumbered}

<div id="refs"></div>

# Appendix {.unlisted .unnumbered}
  
```{r SessionInfo}
pander::pander(sessionInfo())

```
