---
title: "Project: Identification of a Notch transcriptomic signature for breast cancer"
subtitle: "Part 1 - Training Data: Basal-like treated BRCA Cell Lines"
author: "Computational Oncology: Felix Geist"
date: "`r Sys.Date()`"
output: 
  html_document:
    theme: flatly
    highlight: tango
    fig_caption: TRUE
    number_sections: FALSE
    code_folding: hide
    toc: TRUE
    toc_depth: 1
    toc_float:
      collapsed: FALSE
      smooth_scroll: FALSE
bibliography: "`r here::here('src', 'part1a.bib')`"
link-citations: true
biblio-style: "apalike"
---

```{r 00_setup, include=FALSE}

knitr::opts_chunk$set(
	fig.height = 4,                  # 4 inch high, 6 inch wide
	fig.width = 6,
	dpi = 300,                       
	dev = c("svg"),                  
	fig.align = "center",
	#fig.path = "../Figures/Part1/",     
	message = FALSE,
	warning = FALSE,
	echo = TRUE,                     
	cache = FALSE
)
```

```{css, echo=FALSE}

text {
  font-family: sans-serif;
}

h1, .h1, h2, .h2, h3, .h3 {
    margin-top: 0.5em;
}

div.box { 
  background-color:#EEEEEE; 
  border: 2px solid #0F69AF;
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

div.obs { 
  background-color:white; 
  border: 2px solid #0F69AF; 
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

body, td {
   font-size: 14px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 14px
}

img {
    max-width: 80%;
}
```

```{r 01-packages, echo=TRUE, results = 'hide'}

library(tidyverse)
library(here)

library(xlsx)

library(ggplot2)
library(ggpubr)
library(ggrepel)
library(circlize)
library(ComplexHeatmap)
library(ggthemes)

library(org.Hs.eg.db)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(Organism.dplyr)
library(msigdbr)

library(DESeq2)
library(vsn)
library(PCAtools)
library(uwot)
library(rstatix)
library(VennDiagram)
library(BiocParallel)
library(ggtree)

# Citations
knitr::write_bib(file = '../src/part1a.bib')

# color schema

merck_colors <- ggthemes::tableau_color_pal("Color Blind")(10)

merck_colors_ext <- colorRamp2(seq(from = 1, to = 40, by = 4), merck_colors[1:10])

subtype_colors <- merck_colors[c(5,9,6)]
names(subtype_colors) <- c("basal", "her2", "luma")

time_colors <- merck_colors[c(10,7)]
names(time_colors) <- c("8h", "72h")

treatment_colors <- merck_colors[c(1,2,3,4)]
names(treatment_colors) <- c("Notch on", "Notch off", "ground state","Notch on/off")

col_heatmap <- c(merck_colors[1], "white", merck_colors[6])
col_heatmap_not0 <- c(merck_colors[1], merck_colors[3], merck_colors[6])

# set standard theme
theme_set(theme_bw())

options(knitr.kable.NA = '')   # This replaces NA in knitr tables to an empty string

```

**Introduction**

We want to identify robust NOTCH signatures for the activation and inactivation of the NOTCH pathway. Therefore, the group of Urban Lendahl at the Karolinska Institutet, Stockholm, Schweden and the group of Dirk Wienke at the Translational Innovation Platform Oncology and Immuno-Oncology at Merck KGaA, Darmstadt, Germany teamed up.  

<div class = "box">
**Research Questions**

*Treatments:*  

We treated six basal-like breast cancer cell lines either with the gamma-secretase inhibitor DAPT or cultivated the cells on immobilized NOTCH ligand Jagged1. Subsequently, the cells were harvested after 8h or 72h of incubation time and their transcriptome analyzed by RNASeq.  

Each condition was treated for both 8h and for 72h:  

* normal (no treatment): "FC"
* NOTCH ON (cultured on immobilised Jagged1 ligand): "Jagged1"
* NOTCH OFF (blocked with GSI): "DAPT"

Of special interest is the comparison: Jagged1 (NOTCH high) vs. DAPT (NOTCH low)  

**Specific goals:**  

Identify a Notch signature:  

* that defines active and hyperactive notch
* that may identify responsive cell lines / patients

</div>
  
**Methods**

The data is imported and normalized following the `DESeq2` pipeline and subsequently QC checked by dimensionality reduction methods and visualization of positive controls. We pre-filter the genes to only use protein coding genes covered by the three patient data sets used in this study (TCGA, METABRIC and Oslo2). Subsequently, differential expression analysis has been performed.  

**Conclusion**  

<div class = "box">

Differentially expression analysis of the training data set (both 8h and 72h) of the six basal-like cell lines resulted in lists of differentially expressed genes. In a next step, differentially expression analysis will be performed for the 8h and 72h timepoint seperately.    

</div>

# Methods and Data Sources

## Methods

**Platform and Report**  
The analysis was performed using R `r paste(R.Version()[c("major", "minor")], collapse = ".")` [@R-base] with the extension of the `tidyverse` [@R-tidyverse; @tidyverse2019] (dplyr, forcats, ggplot2, purrr, readr, stringr, tibble, tidyr).   
The report was generated using `Rmarkdown` [@R-rmarkdown; @rmarkdown2018; @rmarkdown2020] and `knitr` [@R-knitr; @knitr2014; @knitr2015].
Data was sourced from the X-Omics Platform using the `ProfilerAPI2` [@R-ProfilerAPI2] and `xopdata` [@R-xopdata]. Communication with excel files was performed using the `xlsx` package [@R-xlsx]. 

**Expression Profiling**  
The RNASeq data set and data from public ressources was analyzed using `DESeq2` [@R-DESeq2]. Gene symbols were annotated using the `AnnotationDbi` [@R-AnnotationDbi; @R-AnnotationFilter], `org.Hs.eg.db` [@R-org.Hs.eg.db] and `TxDb.Hsapiens.UCSC.hg38.knownGene` [@R-TxDb.Hsapiens.UCSC.hg38.knownGene] by using `Organism.dplyr` [@R-Organism.dplyr].  
Dimensionality of the data set was reduced by `PCAtools` [@R-PCAtools] and UMAPs generated by `uwot` [@R-uwot]. Notch pathway genes were identified by querying wikipathways using the `msigdbr` package [@R-msigdbr].  

**Visualization**  
Plots are drawn by `ggplot2`, `ggthemes` [@R-ggthemes], `ggpubr` [@R-ggpubr], `ggrepel` [@R-ggrepel] and `cowplot` [@R-cowplot].  Tables are drawn, using the `knitr` package, together with `kableExtra` styling [@R-kableExtra]. VennDiagrams were drawn using the `VennDiagram` package [@R-VennDiagram]. Heatmaps are drawn using the `ComplexHeatmap` package [@R-ComplexHeatmap; @ComplexHeatmap2016].     

# Dataimport

Here, we make use of the count and TPM values that were provided by SciLife labs:  

```{r 02a-dataimport, echo = T, results = 'hide'}

# --- Part 2: Data ingestion and QC

  salmon.merged.gene_counts <- readRDS("../data/Training/TNBC_salmon.merged.gene_counts.rds")

# --- metadata

  lib.path  = base::file.path('../',
                              'data',
                              'Training',
                              'U.Lendahl_21_01_library_info.txt')
  
  met.lib_all   = vroom::vroom(lib.path)
  
  met.lib <- met.lib_all %>%
    dplyr::filter(`Lib QC` != 'FAILED')                       %>%   # redone in library B , except for one sample 
    dplyr::select(sample_name=`NGI ID`,
                  condition_lib_prep=`Lib Prep`)

  # add failed but QC check acceptable library
  
  met.lib <- met.lib %>%
    bind_rows(met.lib_all %>%
                dplyr::filter(!met.lib_all$`NGI ID` %in% met.lib$sample_name) %>%
                dplyr::select(sample_name=`NGI ID`,
                  condition_lib_prep=`Lib Prep`))
  

# --- Read sample metadata
  
  sam.path  = base::file.path('..',
                              'data',
                              'Training',
                              'U.Lendahl_21_01_sample_info.txt')
  
  met.sam = read_delim(sam.path, 
                         delim = "\t", escape_double = FALSE, 
                         trim_ws = TRUE) %>%
    dplyr::select(sample_name=`NGI ID`, condition=`User ID`)      %>%
    tidyr::extract(condition,
                   into=c('condition_cell_line',
                          'condition_treatment',
                          'condition_time',
                          'condition_replicate'),
                   regex='^([a-zA-Z0-9-]+) (.*) ([0-9]+h) \\#([0-9]{1}$)')

  # unique(met.sam$condition_cell_line)
  # [1] "HCC1187"  "MDAMB468" "HCC70"    "HCC1599"  "MDAMB436" "BT20"
  
# --- Read in additional sample metadata
  
  add.met.sam <- read_csv("../data/Training/sample_information.csv") %>% 
    dplyr::select("sample_name" = Sample,
                  "meta_rin" = Rin, 
                  "meta_lib_conc" = `Library Concentration`,
                  "meta_lib_size_bp" = `Average Size bp`,
                  "meta_prep_status" = `Prep Status`)
  
# --- Read in subtypes
# --- Source: Dai et al., J Cancer, 8(16):3131â€“3141 (2017)
# --- TNBCA/B: TNBC-basal A is more luminal-like and TNBC-basal B more basal-like

  met.sub   = base::data.frame(condition_cell_line=c('HCC1599','MDAMB436', 'HCC70',
                                                     'BT20', 'MDAMB468', 'HCC1187'),
                               condition_subtype= c('basal', 'basal', 'basal',
                                                    'basal', 'basal', 'basal'),
                               condition_sub_subtype = c('TNA', 'TNA', 'TNA', 
                                                         'TNA', 'TNA', 'TNA'))
  
# --- Define plan for combining input data

  # --- Additional failed QC -> NONE
  # met.fail = c()
  
  # --- Prepare combined, long quantitation dataframe
  meta.all  <- met.lib %>%
    dplyr::left_join(met.sam,
                      by = "sample_name")                     %>%
    dplyr::left_join(met.sub,
                     by = "condition_cell_line")              %>% 
    dplyr::left_join(add.met.sam,
                     by = "sample_name")                    # %>%
   # dplyr::filter(!sample_name %in% met.fail)
    
  # Overview over replicates of cell lines
    
  meta.all_summary <- meta.all %>% 
    dplyr::select(sample_name, condition_cell_line, condition_treatment, condition_replicate, condition_time) %>% 
    dplyr::distinct() %>% 
    dplyr::group_by(condition_cell_line, condition_treatment, condition_time) %>% 
    dplyr::mutate(`number per group` = n()) %>% 
    dplyr::arrange(condition_cell_line, condition_treatment, desc(condition_time)) %>%
    dplyr::ungroup()
  
  meta.all_summary <- meta.all_summary %>% 
    dplyr::left_join(add.met.sam,
                     by = "sample_name")

  ####### using Organism.dplyr #######
  #
  # new dbdplyr does not work with this package
# 
#   if(!file.exists("../data/part1_qnt.ann.rds")){
#   
#       # --- Add chromosomal annotations to ensembl ids (remove version number)
#     qnt.ann = Organism.dplyr::src_organism('TxDb.Hsapiens.UCSC.hg38.knownGene', overwrite = TRUE) %>%
#       Organism.dplyr::select(keytype='ensembl',
#                              keys=str_remove(unique(rowData(salmon.merged.gene_counts)$gene_id),"\\.[0-9]+$"),
#                              columns=c('symbol', 'gene_chrom', 'genename', 'ensembl', 
#                                        'entrez'))                  %>%
#       dplyr::distinct(symbol, .keep_all=T)  
#     
#     # if duplicated, use the smalles entrez id
#     
#     qnt.ann <- qnt.ann %>%
#       dplyr::group_by(ensembl) %>%
#       dplyr::filter(entrez == min(entrez))
#     
#     saveRDS(qnt.ann, 
#         file = "../data/part1_qnt.ann.rds")
#     
#   } else {
#     qnt.ann <- readRDS("../data/part1_qnt.ann.rds")
#   }
  
  # --- use org.Hs.eg.db

qnt.ann <- AnnotationDbi::select(org.Hs.eg.db, 
                        keys=str_remove(unique(rowData(salmon.merged.gene_counts)$gene_id),"\\.[0-9]+$"), 
                        columns=c('SYMBOL', 'GENENAME', 'ENSEMBL', 'ENTREZID'), 
                        keytype='ENSEMBL') %>%
  dplyr::rename(symbol=SYMBOL, 
                gene_chrom=GENENAME, 
                ensembl=ENSEMBL,
                entrez=ENTREZID) %>%
  dplyr::distinct(symbol, .keep_all=TRUE) %>%
  dplyr::right_join(data.frame(ensembl=str_remove(unique(rowData(salmon.merged.gene_counts)$gene_id),"\\.[0-9]+$")), by='ensembl') 

# if duplicated, use the smalles entrez id
    
    qnt.ann <- qnt.ann %>%
      dplyr::group_by(ensembl) %>%
      dplyr::filter(entrez == min(entrez))
  
  # --- Add information to summarized experiment:
  
  se_counts <- SummarizedExperiment(assays = list(counts = assay(salmon.merged.gene_counts, "counts"),
                                                  abundance = assay(salmon.merged.gene_counts, "abundance")),
                                    rowData = rowData(salmon.merged.gene_counts) %>% 
                                      as.data.frame() %>%
                                      dplyr::select(-gene_name) %>%
                                      dplyr::mutate(ensembl = str_remove(gene_id,"\\.[0-9]+$")) %>%
                                      dplyr::left_join(qnt.ann,
                                                by = c("ensembl")) %>%
                                      column_to_rownames("gene_id"),
                                    colData = colData(salmon.merged.gene_counts) %>%
                                      as.data.frame() %>%
                                      dplyr::mutate(names = str_remove(names, "_R1$")) %>%
                                      dplyr::left_join(meta.all,
                                                       by = c("names" = "sample_name")))
  
  colData(se_counts)$condition_treatment <- factor(colData(se_counts)$condition_treatment, levels = c("FC", "DAPT", "Jagged1"))
  colData(se_counts)$condition_time <- factor(colData(se_counts)$condition_time, levels = c("8h", "72h"))
  colData(se_counts)$condition_cell_line <- factor(colData(se_counts)$condition_cell_line)
  
```

## Data Overview

```{r 02b-dataimport}

  DT::datatable(meta.all_summary,
            class = 'cell-border stripe', 
            extensions = 'Buttons',
            rownames = TRUE,
            options = list(
              scrollX='600px',
              dom = 'Bfrtip',
              buttons = c('excel'),
              headerCallback = DT::JS(
                "function(thead) {",
                "  $(thead).css('font-size', '0.7em');",
                "}"
                )
              )
            ) %>%
    DT::formatStyle(columns = c(1:ncol(meta.all_summary)), fontSize = '70%')
```

# Pre-processing and DESeq2 
  
The gene universe of the intersection of genes annotated in TCGA-BRCA, Oslo2 and METABRIC will be used as a basis for subsequent analyses. In addition:  
Lowly expressed genes, immunoglobulin gene segments, readthrough transcripts, ribosomal genes, sex-linked genes (Y chromosome and Xist), uncharactarized locations, uncharactarized long intergenic non-protein coding RNA and obsolete Entrez Gene IDs were filtered. Protein coding genes, antisense RNA, characterized long non-coding RNA, micro RNAs and chromatin open reading frames were not filtered for.  

The numerical metadata columns used in the analysis are centered and scaled.  

Subsequently, we will make use of the `DESeq2` [@R-DESeq2] pipeline to identify differentially expressed genes.  

```{r 03-DESEQ2}

se_counts_filtered <- se_counts

# load TCGA-BRCA, Oslo2 and METABRIC gene universe, saved in "TCGA_METABRIC_Oslo2_Gene_Universe.rds"
# intersection of genes present in all three datasets: TCGA-BRCA, METABRIC and Oslo2

Gene_universe <- readRDS("../data/TCGA_METABRIC_Oslo2_Gene_Universe.rds")
Gene_universe <- Gene_universe %>%
    mutate(id_ensembl_gene_ids = strsplit(as.character(id_ensembl_gene_ids), "\\|")) %>%
    unnest(id_ensembl_gene_ids)
ridx <- rowData(se_counts_filtered)$ensembl %in% Gene_universe$id_ensembl_gene_ids

# table(ridx)
# FALSE  TRUE 
# 46228 14441 

se_counts_filtered <- se_counts_filtered[ridx,]

# remove lowly expressed genes (less than 10 counts per row)

ridx <- rowSums(assay(se_counts_filtered, "counts")) <= 10
# table(ridx)
# FALSE  TRUE 
# 14085   356 
se_counts_filtered <- se_counts_filtered[!ridx,]

# remove genes with no associated entrez ID

ridx <- is.na(rowData(se_counts_filtered)$entrez)
# table(ridx)
# FALSE  TRUE 
# 14009    76 
se_counts_filtered <- se_counts_filtered[!ridx,]

# remove sex-linked genes (Y chromosome and Xist)

ridx <- rowData(se_counts_filtered)$gene_chrom == "chrY"
ridx <- ifelse(is.na(ridx), FALSE, ridx)
ridx <- ifelse(rowData(se_counts_filtered)$symbol == "XIST", TRUE, ridx)
# table(ridx)
# FALSE  TRUE 
# 14006     3 
se_counts_filtered <- se_counts_filtered[!ridx,]

# remove ribosomal genes

ridx <- grepl(pattern = "^RPS|^RNA5|^RPL", x = rowData(se_counts_filtered)$symbol)
# table(ridx)
# FALSE  TRUE 
# 13910    96 
se_counts_filtered <- se_counts_filtered[!ridx,]

# immuneglobilin genes:

ridx <- grepl(pattern = "^IG[HKL]", x = rowData(se_counts_filtered)$symbol)
# table(ridx)
# FALSE  TRUE 
# 13907     3  
se_counts_filtered <- se_counts_filtered[!ridx,]

# exclude readthrough transcripts:

ridx <- grepl(pattern = "readthrough", x = rowData(se_counts_filtered)$symbol)
# table(ridx)
# FALSE 
# 13907   
se_counts_filtered <- se_counts_filtered[!ridx,]

# uncharactarized locations:

ridx <- grepl(pattern = "^LOC[0-9]+", x = rowData(se_counts_filtered)$symbol)
# table(ridx)
# FALSE  TRUE 
# 13858    49 
se_counts_filtered <- se_counts_filtered[!ridx,]

# long intergenic non-protein coding RNA

ridx <- grepl(pattern = "^LINC[0-9]+", x = rowData(se_counts_filtered)$symbol)
# table(ridx)
# FALSE  TRUE 
# 13844    14  
se_counts_filtered <- se_counts_filtered[!ridx,]

# removed duplicated genes -> take smallest entrez id:

# table(duplicated(rowData(se_counts_filtered)$symbol))
# FALSE 
# 13844 

# scale numerical metadata columns: meta_rin, meta_lib_conc, meta_lib_size_bp

colData(se_counts_filtered)$meta_rin <- as.numeric(scale(colData(se_counts_filtered)$meta_rin))
colData(se_counts_filtered)$meta_lib_conc <- as.numeric(scale(colData(se_counts_filtered)$meta_lib_conc))
colData(se_counts_filtered)$meta_lib_size_bp <- as.numeric(scale(colData(se_counts_filtered)$meta_lib_size_bp))

### --- Define the model design

# export final gene universe:

rnaseq_gene_universe <- as.data.frame(rowData(se_counts_filtered))

saveRDS(rnaseq_gene_universe,
        file = "../data/TNBC_gene_universe.rds")

ddsSE_preparation <- DESeqDataSetFromMatrix(countData = round(assay(se_counts_filtered, "counts")),
                                colData = colData(se_counts_filtered),
                                rowData = rowData(se_counts_filtered),
                                design = ~ 0 + condition_treatment + condition_time + condition_cell_line + 
                                  condition_lib_prep + meta_rin + meta_lib_size_bp)

# DESeq:

ddsSE <- DESeq(ddsSE_preparation)

# Export DESeq2 object for GEO upload

base::saveRDS(ddsSE,
        file = "../data/Training/TNBC_DESeq2.rds")

```

# QC

For unsupervised clustering and PCA analyses, data will be variance stabilized.   

```{r 04a-DESEQ2-QC-meanSD}

###################################################################
#                                                                 #
# Import DESeq2 object from GEO                                   #
#                                                                 #
###################################################################

ddsSE <- base::readRDS("C:/Users/M286901/Desktop/Notch Project/data/Training/TNBC_DESeq2.rds")

# this gives log2(n + 1)
ntd <- normTransform(ddsSE)
p <- meanSdPlot(assay(ddsSE),
                plot = FALSE)
```

```{r 04b-DESEQ2-QC-meanSD}
p$gg +
  ggtitle("Diagnostic Plot: Standard deviations vs row means [log2(n+1)]")
```

```{r 04c-DESEQ2-QC-meanSD}
# variance stabilized
vsd <- vst(ddsSE, blind=FALSE)
p <- meanSdPlot(assay(vsd),
                plot = FALSE)
```

```{r 04d-DESEQ2-QC-meanSD}
p$gg +
  ggtitle("Diagnostic Plot: Standard deviations vs row means [variance stabilized (vst)]")

# rlog
# rld <- rlog(ddsSE, blind=FALSE)
# meanSdPlot(assay(rld))

```

## Diagnostic Plots

* sample quality
* PCA  
* Screeplots  
* Eigenplots  
* Pairplots

```{r}

### --- Sample Quality

ggplot(meta.all_summary,
       aes(x=condition_cell_line,
           y=meta_rin)) +
  geom_point() +
  ggtitle("RIN values by cell line") +
  labs(y = "RIN values",
       x = "Cell Lines")

ggplot(meta.all_summary,
       aes(x=condition_cell_line,
           y=meta_lib_conc)) +
  geom_point() +
  ggtitle("Library Concentrations by cell line") +
  labs(y = "library concentration [nM]",
       x = "Cell Lines")


ggplot(meta.all_summary,
       aes(x=condition_cell_line,
           y=meta_lib_size_bp)) +
  geom_point() +
  ggtitle("Average library size by cell line") +
  labs(y = "average library size [bp]",
       x = "Cell Lines")
```

```{r 5a-PCA-correlations, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 12, fig.height=4}

############################# Principal Component analysis ############################# 

qnt.met_PCA <- colData(vsd) %>%
  as.data.frame() %>%
  dplyr::select(-c("files", "names", "sizeFactor"))

qnt.met_PCA$condition_treatment <- factor(qnt.met_PCA$condition_treatment, levels = c("DAPT", "FC", "Jagged1"),
                                      labels = c("Notch off", "ground state", "Notch on"))

qnt.met_PCA$Treatment <- paste(qnt.met_PCA$condition_treatment, "-",qnt.met_PCA$condition_time)

qnt.met_PCA$sample_name <- rownames(qnt.met_PCA)

p <- pca(assay(vsd), metadata = qnt.met_PCA, removeVar = 0.1)

horn <- parallelPCA(assay(vsd))
#horn$n

elbow <- findElbowPoint(p$variance)
#elbow

eigencorplot(p,
             components = getComponents(p, 1:20),
             metavars = c('condition_lib_prep','condition_cell_line','condition_treatment', 'condition_time', 'condition_replicate',
                          'meta_rin', "meta_lib_conc", "meta_lib_size_bp"),
             col = c('white', merck_colors[5], merck_colors[1]),
             cexCorval = 0.7,
             colCorval = 'black',
             fontCorval = 2,
             posLab = 'bottomleft',
             posColKey = 'right',
             rotLabX = 45,
             cexLabColKey = 1.5,
             scale = TRUE,
             main = 'PC1-20 correlations',
             colFrame = 'white',
             plotRsquared = TRUE)


```

Numbers of genes used for the PCA: **`r length(p$xvars)`**

These results indicate, that we will can add the additional metadata sample information RIN and average library size in bp to improve our model. The concentration of the library had only minor influence on the total sample variance.  

```{r 05b-PCA-screeplot, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 4, fig.height=3}

screeplot(p,
          components = getComponents(p, 1:20),
          vline = c(horn$n, elbow),
          colBar = merck_colors[1],
          colCumulativeSumLine = merck_colors[6]) +
  theme_classic() + 
  geom_text(aes(horn$n - 1.5, 30, label = "Horn's", vjust = -1)) +
  geom_text(aes(elbow + 1.5, 30, label = "Elbow", vjust = -1)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

```

### PCA

```{r 05c-PCA-1, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 6, fig.height=5}

biplot(p,
       title = "PC1 vs PC2",
       subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))


biplot(p,
       title = "PC1 vs PC2",
       subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'Treatment',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right',
       ellipse = TRUE,
       ylim = c(-150,150),
       xlim = c(-150,150)) +
  scale_shape_manual(values = c(15,16,17,18,19, 20))
```


```{r 05c-PCA-1b, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 6, fig.height=5}

biplot(p,
       title = "PC6 vs PC7",
       x = "PC6",
       y = "PC7",
       subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))
```


```{r 05c-PCA-1c, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 6, fig.height=5}

biplot(p,
       title = "PC6 vs PC10",
       x = "PC6",
       y = "PC10",
       subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))
```


```{r 05c-PCA-1d, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 6, fig.height=5}

biplot(p,
       title = "PC7 vs PC10",
       x = "PC7",
       y = "PC10",
       subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))
```


```{r 05c-PCA-1e, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 17, fig.height=15}

pairsplot_plot <- pairsplot(p,
          trianglelabSize = 9,
          pointSize = 0.5,
          legendLabSize = 3,
          legendIconSize = 0.5,
          labSize = 0.75,
          axisLabSize = 5,
          titleLabSize = 16,
          hlineWidth = 0.2,
          vlineWidth = 0.2,
          borderWidth = 0.4,
          margingaps = unit(c(0.05, 0.05, 0.05, 0.05), "cm"),
       title = "Pairsplot",
       components = getComponents(p, seq_len(10)),
       #subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'none') +
  scale_shape_manual(values = c(0,15,1,16,2,17))

pairsplot_plot
```

```{r 05c-PCA-2, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 10, fig.height=9}

biplot(p,
       title = "PC1 vs PC2",
       subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))

biplot(p,
       title = "PC1 vs PC2",
       subtitle = "Cell Line and time point",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "condition_time",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right')

biplot(p,
       title = "PC1 vs PC2",
       subtitle = "Library Prep Batch and treatment condition",
       lab = NA,
       colby = 'condition_lib_prep',
       shape = "Treatment",
       colkey = as.character(merck_colors[c(8,4)]),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))

biplot(p,
       x = "PC6",
       y = "PC11",
       title = "PC6 vs PC11",
       subtitle = "Treatment and Time",
       lab = NA,
       colby = 'condition_time',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right')  +
  scale_shape_manual(values = c(0,15,1,16,2,17))

biplot(p,
       x = "PC5",
       y = "PC11",
       title = "PC5 vs PC11",
       subtitle = "Cell Line and Treatment",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))



biplot(p,
       x = "PC10",
       y = "PC11",
       title = "PC10 vs PC11",
       subtitle = "Cell Line and treatment condition",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))
```

### UMAP

```{r 05d-Umap, echo=TRUE, message=FALSE, warning=FALSE, fig.width = 7, fig.height = 6}

z_umap <- umap(scale(assay(vsd)) %>% na.omit() %>% t, pca = 20)

z_umap_df <- data.frame(umap1 = z_umap[,1],
           umap2 = z_umap[,2],
           qnt.met_PCA,
           stringsAsFactors = FALSE)

rownames(z_umap_df) <- rownames(qnt.met_PCA)

ggscatter(z_umap_df,
          x = "umap1",
          y = "umap2",
          #label = "condition_cell_line",
          #repel = TRUE,
          color = "condition_cell_line",
          shape = "Treatment",
          #alpha = "breast_alpha",
          size = 2,
          legend = "right",
          palette = as.character(merck_colors),
          title = "UMAP Dimensionality Reduction") +
  geom_text_repel(data = z_umap_df %>% group_by(condition_cell_line) %>% 
               summarise(umap1 = mean(umap1), 
                         umap2 = mean(umap2)),
             aes(x = umap1,
                 y = umap2,
                 label = condition_cell_line,
                 color = condition_cell_line)) + 
  guides(col = guide_legend(nrow = 7)) +
  scale_shape_manual(values = c(0,15,1,16,2,17))

z_tumap <- tumap(scale(assay(vsd)) %>% na.omit() %>% t, pca = 20)

z_tumap_df <- data.frame(tumap1 = z_tumap[,1],
           tumap2 = z_tumap[,2],
           qnt.met_PCA,
           stringsAsFactors = FALSE)

rownames(z_tumap_df) <- rownames(qnt.met_PCA)

ggscatter(z_tumap_df,
          x = "tumap1",
          y = "tumap2",
          #label = "condition_cell_line",
          #repel = TRUE,
          color = "condition_cell_line",
          shape = "Treatment",
          #alpha = "breast_alpha",
          size = 2,
          legend = "right",
          palette = as.character(merck_colors),
          title = "t-UMAP Dimensionality Reduction") +
  geom_text_repel(data = z_tumap_df %>% group_by(condition_cell_line) %>% 
               summarise(tumap1 = mean(tumap1), 
                         tumap2 = mean(tumap2)),
             aes(x = tumap1,
                 y = tumap2,
                 label = condition_cell_line,
                 color = condition_cell_line)) + 
  guides(col = guide_legend(nrow = 7)) +
  scale_shape_manual(values = c(0,15,1,16,2,17))

save.image("Part1image.RData")

```

## Dendrograms of Basal-like cell line panel

### Pearson Correlation

```{r fig.height=8, fig.width=4}

condition_labels <- data.frame(files = colnames(assay(vsd))) %>% 
  dplyr::left_join(as.data.frame(colData(vsd)),
                   by = join_by(files)) %>% 
  dplyr::mutate(condition_label_name = case_when(
    condition_treatment == "FC" ~ paste0(condition_cell_line, "_ground_state_", condition_time, "_", condition_replicate),
    condition_treatment == "DAPT" ~ paste0(condition_cell_line, "_notch_off_", condition_time, "_", condition_replicate),
    condition_treatment == "Jagged1" ~ paste0(condition_cell_line, "_notch_on_", condition_time, "_", condition_replicate)
  ))

vsd_matrix <- assay(vsd)
colnames(vsd_matrix) <- str_replace_all(condition_labels$condition_label_name, "\\_", " ")

# Calculate the distance matrix
dist_matrix <- as.dist(1 - cor(vsd_matrix, method = "pearson"))

# Perform hierarchical clustering
hc_pearson <- hclust(dist_matrix, method = "ward.D2")

# Convert hclust to ggtree object
p <- ggtree::ggtree(hc_pearson,
                    layout = "rectangular") + 
  geom_tippoint(color=merck_colors[1], shape=17, size=2) + 
  coord_cartesian(clip = 'off') + 
  geom_tiplab(size = 1.5,
              offset = 0.01) +
  theme_tree(plot.margin=margin(6, 120, 6, 5))+
  labs(title = "Pearson Distance between the samples with ward.D2 clustering",
       subtitle = "6 basal-like cell line panel")
p

# Perform hierarchical clustering
hc_pearson <- hclust(dist_matrix, method = "complete")

# Convert hclust to ggtree object
p <- ggtree::ggtree(hc_pearson,
                    layout = "rectangular") + 
  geom_tippoint(color=merck_colors[1], shape=17, size=2) + 
  coord_cartesian(clip = 'off') + 
  geom_tiplab(size = 1.5,
              offset = 0.01) +
  theme_tree(plot.margin=margin(6, 120, 6, 5))+
  labs(title = "Pearson Distance between the samples with ward.D2 clustering",
       subtitle = "6 basal-like cell line panel")
p
```


```{r fig.height=8, fig.width=4}
# Calculate the Euclidean distance matrix
dist_matrix_euclidean <- dist(t(vsd_matrix), method = "euclidean")

# Hierarchical clustering
hc_euclidean <- hclust(dist_matrix_euclidean, method = "ward.D2")

# Convert hclust to ggtree object
p <- ggtree::ggtree(hc_euclidean,
                    layout = "rectangular") + 
  geom_tippoint(color=merck_colors[1], shape=17, size=2) + 
  coord_cartesian(clip = 'off') + 
  geom_tiplab(size = 1.5,
              offset = 10) +
  theme_tree(plot.margin=margin(6, 120, 6, 5))+
  labs(title = "Euclidean Distance between the samples with ward.D2 clustering",
       subtitle = "6 basal-like cell line panel")
p

# Hierarchical clustering
hc_euclidean <- hclust(dist_matrix_euclidean, method = "ward.D2")

# Convert hclust to ggtree object
p <- ggtree::ggtree(hc_euclidean,
                    layout = "rectangular") + 
  geom_tippoint(color=merck_colors[1], shape=17, size=2) + 
  coord_cartesian(clip = 'off') + 
  geom_tiplab(size = 1.5,
              offset = 10) +
  theme_tree(plot.margin=margin(6, 120, 6, 5))+
  labs(title = "Euclidean Distance between the samples with complete clustering",
       subtitle = "6 basal-like cell line panel")
p
```


```{r fig.height=8, fig.width=4}
# Calculate the Manhattan distance matrix
dist_matrix_manhattan <- dist(t(vsd_matrix), method = "manhattan")

# Hierarchical clustering
hc_manhattan <- hclust(dist_matrix_manhattan, method = "ward.D2")

# Convert hclust to ggtree object
p <- ggtree::ggtree(hc_euclidean,
                    layout = "rectangular") + 
  geom_tippoint(color=merck_colors[1], shape=17, size=2) + 
  coord_cartesian(clip = 'off') + 
  geom_tiplab(size = 1.5,
              offset = 10) +
  theme_tree(plot.margin=margin(6, 120, 6, 5))+
  labs(title = "Manhattan Distance between the samples with ward.D2 clustering",
       subtitle = "6 basal-like cell line panel")
p


# Hierarchical clustering
hc_manhattan <- hclust(dist_matrix_manhattan, method = "complete")

# Convert hclust to ggtree object
p <- ggtree::ggtree(hc_euclidean,
                    layout = "rectangular") + 
  geom_tippoint(color=merck_colors[1], shape=17, size=2) + 
  coord_cartesian(clip = 'off') + 
  geom_tiplab(size = 1.5,
              offset = 10) +
  theme_tree(plot.margin=margin(6, 120, 6, 5))+
  labs(title = "Manhattan Distance between the samples with complete clustering",
       subtitle = "6 basal-like cell line panel")
p


```

## Heatmap of variable genes

Here, we analyze the top 1000 variable genes and cluster the samples by average using manhatten distance.  

```{r 06-hm-top-variant-genes, fig.width=8, fig.height=8, dev = "png"}

ntop <- order(matrixStats::rowVars(assay(vsd)), decreasing = TRUE)[1:1000]
cpm_top <- counts(ddsSE,normalized=TRUE)[ntop,]
colnames(cpm_top) <- qnt.met_PCA$condition_cell_line[colnames(cpm_top) %in% qnt.met_PCA$sample_name]

Subtype <- as.character(qnt.met_PCA$condition_subtype)
Subtype_col <- subtype_colors[levels(qnt.met_PCA$condition_subtype)]

# lib_prep <- as.character(qnt.met_no_ctrl$condition_lib_prep)
# lib_prep_col <- merck_colors[c(4,5)]
# names(lib_prep_col) = unique(lib_prep)

cell_line <- as.character(qnt.met_PCA$condition_cell_line)
cell_line_col <- merck_colors[1:length(unique(cell_line))]
names(cell_line_col) = unique(cell_line)

treatment_time <- as.character(qnt.met_PCA$condition_time)
treatment_time_col <- time_colors[levels(qnt.met_PCA$condition_time)]
#names(treatment_time_col) = unique(treatment_time)

treatment_factor <- qnt.met_PCA$condition_treatment
treatment <- as.character(treatment_factor)
treatment_col <- treatment_colors[levels(treatment_factor)]

col_fun = colorRamp2(c(8, 4, 0), col_heatmap)

col_ha = ComplexHeatmap::columnAnnotation(#Subtype = Subtype,
                                          #`Library Prep` = lib_prep,
                                          `Cell line` = cell_line,
                                          `Time point` = treatment_time,
                                          Treatment = treatment,
                                          col = list(#Subtype = Subtype_col,
                                                     #`Library Prep` = lib_prep_col,
                                                     `Cell line` = cell_line_col,
                                                     `Time point` = treatment_time_col,
                                                     Treatment = treatment_col),
                                          annotation_name_side = "right")

ht <- Heatmap(cpm_top,
        column_title = "Top 1000 variant genes in the breast cancer cell lines",
        cluster_rows = TRUE,
        cluster_columns = TRUE,
        clustering_method_rows = "average",
        clustering_distance_rows = "manhattan",
        clustering_method_columns = "average",
        clustering_distance_columns = "manhattan",
        #clustering_distance_rows = "euclidean",
        #clustering_distance_columns = "euclidean",
        col = col_fun,  
        show_row_names = FALSE,
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 4),
        top_annotation = col_ha,
        heatmap_legend_param = list(
           title = "log2[cpm]"#,
           #title_position = "lefttop-rot"
           )
        )

draw(ht)
# , merge_legend = TRUE, adjust_annotation_extension = TRUE
```

# Diagnositic Plots

## Known Notch targets

Diagnostic plots of the TPM values of known targets of the Notch pathway are used to validate the experimental conditions.  

```{r 07-Diagnostic-Plots, fig.width=8, fig.height=8}

# --- Read in pre-processed data

# --- Get TPM Values and exclude QC failed datapoints

salmon_merged_gene_tpm <- read_delim("../data/Training/salmon.merged.gene_tpm.tsv", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

# subset genes by the categories stated before and add gene information

salmon_merged_gene_tpm <- salmon_merged_gene_tpm %>%
  dplyr::mutate(ensembl = str_remove(gene_id, "\\.[0-9]+$")) %>%
  dplyr::inner_join(rowData(se_counts) %>%
                     as_tibble() %>%
                     dplyr::select(symbol, ensembl, entrez),
                   by = "ensembl")

salmon_merged_gene_tpm <- salmon_merged_gene_tpm %>%
  tidyr::pivot_longer(-c(gene_id, symbol, ensembl, entrez),
                      names_to = "sample_name",
                      values_to = "quant_tpm")

# sum up TPM by gene -> already done
# 
# salmon_merged_gene_tpm <- salmon_merged_gene_tpm %>%
#   dplyr::group_by(gene_id, symbol, ensembl, entrez, sample_name) %>%
#   dplyr::summarise(quant_tpm = sum(quant_tpm)) %>%
#   dplyr::ungroup()

qnt.mat_tpm = salmon_merged_gene_tpm  %>%
  mutate(quant_log2tpm = log2(quant_tpm+1))

# add cell line information:

qnt.mat_tpm_agg_log2 <- qnt.mat_tpm %>%
  dplyr::left_join(colData(se_counts) %>%
                     as_tibble(),
                   by = c("sample_name" = "files"))

# save TPM object

saveRDS(qnt.mat_tpm_agg_log2, file = "../results/Training/TNBC_TPM.rds")

############################# Diagnostic Plots ############################# 

diag_plot <- function(gene) {
  plotdata <- qnt.mat_tpm_agg_log2 %>%
  filter(symbol == !!gene) %>%
  dplyr::mutate(condition_treatment = dplyr::recode_factor(condition_treatment,
                                                           `DAPT`      = 'Notch off',
                                                           `FC`   = 'ground state',
                                                           `Jagged1`     = 'Notch on'),
                condition_time = factor(condition_time, levels = c("8h", "72h")))

  p <- ggplot(plotdata,
         aes(y = quant_log2tpm,
             x = condition_time,
             fill = condition_treatment)) +
    geom_boxplot(position=position_dodge(1), outlier.colour = NA) +
    theme_pubclean() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    geom_point(size = 2, position=position_dodge(1)) +
    ylim(c(0,range(plotdata$quant_log2tpm)[2])) +
    facet_wrap(condition_cell_line~condition_subtype, 
               ncol=3,
               scales = "free")+
    labs(title = gene,
         x = "Condition",
         y = "log(TPM+1)") +
   # scale_fill_manual(values = time_colors[levels(factor(plotdata$condition_time))])
    scale_fill_manual(values = treatment_colors[levels(factor(plotdata$condition_treatment))])
    
    
  print(p)
}

diag_plot("NRARP")
diag_plot("HEY1")
diag_plot("HES1")

```

## Notch expression heatmap 8h + 72h

```{r 08a-notch-expression-hm, fig.height=3.2, fig.width=6}

notch_expression <- qnt.mat_tpm_agg_log2 %>%
  filter(condition_treatment == "FC") %>%
  filter(symbol %in% c("NOTCH1","NOTCH2","NOTCH3","NOTCH4")) %>%
  dplyr::select(symbol, quant_log2tpm, condition_cell_line, condition_time, condition_subtype) %>%
  arrange(symbol, condition_subtype, condition_cell_line, desc(condition_time))

matrix_notch_expression <- notch_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("symbol") 

Subtype <- as.character(notch_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_subtype"))

Subtype_col <- subtype_colors[unique(Subtype)]

Timepoint <- as.character(notch_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_time"))

Timepoint_col <- time_colors[unique(Timepoint)]

col_fun = colorRamp2(c(0, 4, 8), col_heatmap)

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          #Cellline = Cellline,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col
                                                     #Cellline = Cellline_col
                                                     ), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")#,
                                            #Cellline = list(nrow = 4,
                                            #                at = notch_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% pull("condition_cell_line") %>% unique())
                                            )
)

ht <- ComplexHeatmap::Heatmap(matrix_notch_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = notch_expression %>% filter(gene_name == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
)

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          #Cellline = Cellline,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col#,
                                                     #Cellline = Cellline_col
                                                     ), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")#,
                                           # Cellline = list(nrow = 4)
                                          )
)

ht <- ComplexHeatmap::Heatmap(matrix_notch_expression,
                        cluster_rows = FALSE,
                        cluster_columns = TRUE,
                        #column_split  = notch_expression %>% filter(gene_symbol == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
                        )

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)

```

## Endocrine receptor status of cell lines - 8h + 72h

```{r 08c-receptors-expression-hm, fig.height=3.2, fig.width=6}

receptor_expression <- qnt.mat_tpm_agg_log2 %>%
  filter(condition_treatment == "FC") %>%
  filter(symbol %in% c("ERBB2","ERBB3","ESR1","PGR","AR")) %>%
  dplyr::select(symbol, quant_log2tpm, condition_cell_line, condition_time, condition_subtype) %>%
  arrange(symbol, condition_subtype, condition_cell_line, desc(condition_time))

matrix_receptor_expression <- receptor_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("symbol")

Subtype <- as.character(receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_subtype"))

Subtype_col <- subtype_colors[unique(Subtype)]

Timepoint <- as.character(receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_time"))

Timepoint_col <- time_colors[unique(Timepoint)]

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          #Cellline = Cellline,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col
                                                     #Cellline = Cellline_col
                                                     ), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")#,
                                            #Cellline = list(nrow = 4,
                                            #                at = matrix_receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% pull("condition_cell_line") %>% unique())
                                            )
)

ht <- ComplexHeatmap::Heatmap(matrix_receptor_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = notch_expression %>% filter(gene_name == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
)

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          #Cellline = Cellline,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col#,
                                                     #Cellline = Cellline_col
                                                     ), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")#,
                                           # Cellline = list(nrow = 4)
                                          )
)

ht <- ComplexHeatmap::Heatmap(matrix_receptor_expression,
                        cluster_rows = FALSE,
                        cluster_columns = TRUE,
                        #column_split  = notch_expression %>% filter(gene_symbol == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
                        )

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)

```

## Combined plot

```{r 08e-receptors-expression-hm, fig.height=2.5, fig.width=5}

# load wikipathway notch
# Wikipathway

wikipathway_genes <- msigdbr::msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:WIKIPATHWAYS")

# filter notch terms, use Notch Signaling (homo sapiens)

notch_pathways <- wikipathway_genes %>%
  dplyr::filter(grepl("WP_NOTCH_SIGNALING$", gs_name, ignore.case = TRUE))

notch_pathways_genes_entrez <- unique(notch_pathways$human_entrez_gene) 

notch_pathways_genes_symbol <-clusterProfiler::bitr(notch_pathways_genes_entrez, fromType="ENTREZID", toType="SYMBOL", OrgDb="org.Hs.eg.db")

notch_pathways_genes_symbol <- unique(notch_pathways_genes_symbol$SYMBOL)

### CCLE phenotype data
# Phenotype data were downloaded from depmap-CCLE (21q1)
# https://depmap.org/portal/download/all/
# and stored in "data/CCLE/"

CCLE_sample_info_21Q1 <- read_csv("../data/CCLE/CCLE_sample_info_21Q1.csv")

# subset to used cell lines

qnt.mat_tpm_agg_log2$stripped_cell_line_name <- str_remove(toupper(qnt.mat_tpm_agg_log2$condition_cell_line), "-")

qnt.mat_tpm_agg_log2 <- qnt.mat_tpm_agg_log2 %>%
  left_join(CCLE_sample_info_21Q1 %>%
              dplyr::select(stripped_cell_line_name, DepMap_ID),
            by = "stripped_cell_line_name")

CCLE_sample_info_21Q1_used <- unique(qnt.mat_tpm_agg_log2$DepMap_ID) %>%
  na.omit()

### mutation data
# Mutations were downloaded from depmap-CCLE (21q1)
# https://depmap.org/portal/download/all/

CCLE_mutations_csv_21Q1 <- read_csv("../data/CCLE/CCLE_mutations.csv_21Q1.csv")

# remove silent, keep only cell lines of screen
CCLE_mutations_csv_21Q1 <- CCLE_mutations_csv_21Q1 %>%
  dplyr::filter(Variant_annotation != "silent",
                DepMap_ID %in% !!CCLE_sample_info_21Q1_used)

# filter notch mutations:

notch_CCLE_mutations_csv_21Q1 <- CCLE_mutations_csv_21Q1 %>%
  dplyr::filter(Entrez_Gene_Id %in% !!unique(notch_pathways$human_entrez_gene))

# if variant_protein is NA -> take variant_genome and variant_consequence

notch_CCLE_mutations_csv_21Q1$variant <- ifelse(is.na(notch_CCLE_mutations_csv_21Q1$Protein_Change), 
                                               paste0(notch_CCLE_mutations_csv_21Q1$Variant_Classification,"(", notch_CCLE_mutations_csv_21Q1$Genome_Change, ")"),
                                               notch_CCLE_mutations_csv_21Q1$Protein_Change)

notch_CCLE_mutations_csv_21Q1 <- notch_CCLE_mutations_csv_21Q1 %>% 
  dplyr::select(c("variant", "Hugo_Symbol", "DepMap_ID"))

# Aggregate multiple mutations per gene into one string:

notch_CCLE_mutations_csv_21Q1_agg <- aggregate(variant ~ Hugo_Symbol + DepMap_ID, 
                                              data = notch_CCLE_mutations_csv_21Q1, 
                                              FUN = function(x) paste(unique(x), collapse=", "))

notch_CCLE_mutations_csv_21Q1_wide <- pivot_wider(notch_CCLE_mutations_csv_21Q1_agg,
                                      names_from = Hugo_Symbol,
                                      names_prefix = "mutated_",
                                      values_from = variant)

notch_CCLE_mutations_csv_21Q1_wide <- data.frame(DepMap_ID = CCLE_sample_info_21Q1_used,
           stringsAsFactors = FALSE) %>%
  left_join(notch_CCLE_mutations_csv_21Q1_wide)

# replace detailed mutations:

notch_CCLE_mutations_csv_21Q1_wide <- notch_CCLE_mutations_csv_21Q1_wide %>%
  mutate_at(vars(matches("mutated_")), function(x) ifelse(is.na(x), "wt", x))

notch_CCLE_mutations_csv_21Q1_long <- notch_CCLE_mutations_csv_21Q1_wide %>%
  pivot_longer(-DepMap_ID,
               names_to = "gene_symbol",
               values_to = "mutation",
               names_prefix = "mutated_") %>%
  left_join(qnt.mat_tpm_agg_log2 %>%
              dplyr::select(stripped_cell_line_name, DepMap_ID) %>%
              dplyr::distinct())

```

```{r 08f-receptors-expression-hm, fig.height=5, fig.width=6}

receptor_expression <- qnt.mat_tpm_agg_log2 %>%
  filter(condition_treatment == "FC") %>%
  filter(symbol %in% c("ERBB2","ERBB3","ESR1","PGR","AR")) %>%
  dplyr::select(gene_symbol = symbol, quant_log2tpm, condition_cell_line, condition_subtype, condition_time) %>%
  arrange(gene_symbol, condition_subtype, condition_cell_line, desc(condition_time))

matrix_receptor_expression <- receptor_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("gene_symbol")

notch_expression <- qnt.mat_tpm_agg_log2 %>%
  filter(condition_treatment == "FC") %>%
  filter(symbol %in% c("NOTCH1", "NOTCH2", "NOTCH3", "NOTCH4")) %>%
  dplyr::select(gene_symbol = symbol, quant_log2tpm, condition_cell_line, condition_subtype, condition_time) %>%
  arrange(gene_symbol, condition_subtype, condition_cell_line, desc(condition_time))

matrix_notch_expression <- notch_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("gene_symbol") 

notch_CCLE_mutations_use_wide <- notch_expression %>%
  dplyr::select(-gene_symbol, -quant_log2tpm) %>%
  dplyr::distinct() %>%
  left_join(notch_CCLE_mutations_csv_21Q1_long %>%
    dplyr::select(-DepMap_ID),
    by = c("condition_cell_line" = "stripped_cell_line_name")) %>%
  arrange(gene_symbol, condition_subtype, condition_cell_line, desc(condition_time))
  
# xlsx::write.xlsx(notch_CCLE_mutations_use_wide %>%
#                    dplyr::select(-condition_time) %>%
#                    dplyr::distinct() %>%
#                    pivot_wider(names_from = gene_symbol,
#                                values_from = mutation),
#                  file = "../results/NOTCH_pathway_mutations.xlsx")

matrix_notch_mutation <- notch_expression %>% 
  dplyr::select(-condition_subtype, -gene_symbol, -quant_log2tpm) %>%
  dplyr::left_join(notch_CCLE_mutations_use_wide) %>% 
  dplyr::select(-condition_subtype) %>%
  dplyr::distinct() %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = mutation) %>%
  dplyr::filter(!is.na(gene_symbol)) %>%
  column_to_rownames("gene_symbol")

matrix_notch_mutation <- matrix_notch_mutation[rownames(matrix_notch_mutation) %in% c("NOTCH1", "NOTCH2", "NOTCH3", "NOTCH4"),colnames(matrix_receptor_expression)]

# header

Subtype <- as.character(receptor_expression[,c("condition_cell_line", "condition_subtype", "condition_time")] %>% distinct() %>% pull("condition_subtype"))
Subtype_col <- subtype_colors[unique(Subtype)]

Timepoint <- as.character(receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_time"))
Timepoint_col <- time_colors[unique(Timepoint)]

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")
                                            )
)

ht <- ComplexHeatmap::Heatmap(matrix_receptor_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = col_split,
                        row_title = "Expression of\nEndocrine\nReceptors",
                        row_title_rot = 0,
                        row_title_gp = gpar(fontsize = 10),
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "scaled log2(TPM+1)", 
                          #title_position = "leftcenter-rot",
                          direction = "horizontal")
)

ht2 <- ComplexHeatmap::Heatmap(matrix_notch_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = col_split,
                        row_title = "Expression of\nNotch\nReceptors",
                        row_title_rot = 0,
                        row_title_gp = gpar(fontsize = 10),
                        show_column_names = TRUE,
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        show_heatmap_legend = FALSE,
                        heatmap_legend_param = list(
                          title = "scaled log2(TPM+1)", 
                          title_position = "leftcenter-rot")
)

mutation_names <- unique(unlist(matrix_notch_mutation))
mutation_names <- mutation_names[!is.na(mutation_names)]

col_mut = c("lightgrey",merck_colors[1:(length(mutation_names)-1)])
names(col_mut) <- mutation_names


ht3 <- ComplexHeatmap::Heatmap(matrix_notch_mutation,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = col_split,
                        row_title = "Mutations in\nNotch\nReceptors",
                        row_title_rot = 0,
                        row_title_gp = gpar(fontsize = 10),
                        show_column_names = TRUE,
                        col = col_mut,
                        na_col = "white",
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "NOTCH Mutations")
                        
)

draw(ht %v% ht2 %v% ht3, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)

```

# Differential Gene Expression Analysis

We extract the contrasts of interest from the DESEQ2 object, using a FDR cutoff of 0.05.  

## Both Timepoints

### FC vs DAPT

```{r 09a-DE-FC-DAPT, fig.width=4, fig.height=4, dev = "png"}

res_DAPT <- results(ddsSE, contrast=c("condition_treatment","DAPT","FC"), alpha = 0.05)

hist(res_DAPT$pvalue)

#We can order our results table by the smallest p value:

res_DAPTOrdered <- res_DAPT[order(res_DAPT$padj),]

#We can summarize some basic tallies using the summary function.

summary(res_DAPT, alpha = 0.05)

plotMA(res_DAPT, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot - FC vs. DAPT")

printDF_DAPT <- res_DAPT %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(se_counts_filtered) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

  DT::datatable(printDF_DAPT %>% 
          mutate_if(is.numeric, ~signif(., 2)), 
            class = 'cell-border stripe', 
            extensions = 'Buttons',
            rownames = TRUE,
            options = list(
              scrollX='600px',
              dom = 'Bfrtip',
              buttons = c('excel'),
              headerCallback = DT::JS(
                "function(thead) {",
                "  $(thead).css('font-size', '0.7em');",
                "}"
                )
              )
            ) %>%
    DT::formatStyle(columns = c(1:ncol(printDF_DAPT)), fontSize = '70%')

```

#### Vulcaono Plot

```{r 09b-DE-FC-DAPT-vulcano, fig.width=4, fig.height=4}

# resultsNames(ddsSE)
# For visualization of Vulcano Plots, log fold changes are shrunken by "ashr"

resLFC <- lfcShrink(ddsSE, contrast=c("condition_treatment","DAPT","FC"), type="ashr")

# summary(resLFC, alpha = 0.05)

plotMA(resLFC, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot - FC vs. DAPT\nAshr shrinked log2 fold changes")

# first remove the filtered genes (FDR=NA) and create a -log10(FDR) column
filtTab <- resLFC %>%
  as.data.frame() %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(se_counts_filtered) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol),
                   by = "ensembl") %>%
    mutate(`-log10(padj)` = -log10(padj)) %>%
  dplyr::filter(!is.na(`-log10(padj)`))

xlim <- range(filtTab$log2FoldChange)
xlim[1] <- ifelse(xlim[1] > -2, -2, xlim[1])
xlim[2] <- ifelse(xlim[2] < 2, 2, xlim[2])
xlim[1] <- ifelse(abs(xlim[2]) > abs(xlim[1]), -xlim[2], xlim[1])
xlim[2] <- ifelse(abs(xlim[1]) > abs(xlim[2]), -xlim[1], xlim[2])

ggplot(filtTab %>%
         dplyr::filter(!is.na(padj)), 
       aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol)) + 
  theme_classic() + 
  geom_point(aes(colour=padj < 0.05), size=1) +
  coord_cartesian(xlim = xlim) +
  geom_text_repel(data = filtTab %>%
                    dplyr::filter(padj < 0.05,
                                  abs(log2FoldChange) > 0.5),
                  aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol),
                  size = 3,
                  max.overlaps = 20,
                  force = 2) +
  scale_color_manual(values = as.character(merck_colors)[c(1,6)]) +
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed",
             color = "grey") +
  geom_vline(xintercept = 0,
             linetype = "dashed",
             color = "grey") +
  labs(x = "log2 fold Change",
       y = "-log10(p adj)",
       title = "Notch Off (DAPT) vs. Control (FC)")

```

### Jagged1 vs FC

```{r 10a-DE-FC-Jagged1, fig.width=4, fig.height=4, dev = "png"}

res_Jagged1 <- results(ddsSE, contrast=c("condition_treatment","Jagged1","FC"))

hist(res_Jagged1$pvalue)

# We can order our results table by the smallest p value:

res_Jagged1Ordered <- res_Jagged1[order(res_Jagged1$padj),]
#We can summarize some basic tallies using the summary function.

summary(res_Jagged1, alpha = 0.05)

plotMA(res_Jagged1, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. FC", sub = "log2 fold changes")

printDF_Jagged1 <- res_Jagged1 %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(se_counts_filtered) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

  DT::datatable(printDF_Jagged1 %>% 
          mutate_if(is.numeric, ~signif(., 2)), 
            class = 'cell-border stripe', 
            extensions = 'Buttons',
            rownames = TRUE,
            options = list(
              scrollX='600px',
              dom = 'Bfrtip',
              buttons = c('excel'),
              headerCallback = DT::JS(
                "function(thead) {",
                "  $(thead).css('font-size', '0.7em');",
                "}"
                )
              )
            ) %>%
    DT::formatStyle(columns = c(1:ncol(printDF_Jagged1)), fontSize = '70%')
```

#### Vulcaono Plot

```{r 10b-DE-FC-Jagged1-vulcano, fig.width=4, fig.height=4}

# resultsNames
# For visualization of Vulcano Plots, log fold changes are shrunken by "ashr"

resLFC <- lfcShrink(ddsSE, contrast=c("condition_treatment","Jagged1","FC"), type="ashr")

# summary(resLFC, alpha = 0.05)

plotMA(resLFC, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. FC\nAshr shrinked log2 fold changes")

# first remove the filtered genes (FDR=NA) and create a -log10(FDR) column
filtTab <- resLFC %>%
  as.data.frame() %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(se_counts_filtered) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol),
                   by = "ensembl") %>%
    mutate(`-log10(padj)` = -log10(padj)) %>%
  dplyr::filter(!is.na(`-log10(padj)`))

xlim <- range(filtTab$log2FoldChange)
xlim[1] <- ifelse(xlim[1] > -2, -2, xlim[1])
xlim[2] <- ifelse(xlim[2] < 2, 2, xlim[2])
xlim[1] <- ifelse(abs(xlim[2]) > abs(xlim[1]), -xlim[2], xlim[1])
xlim[2] <- ifelse(abs(xlim[1]) > abs(xlim[2]), -xlim[1], xlim[2])

ggplot(filtTab %>%
         dplyr::filter(!is.na(padj)), 
       aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol)) + 
  theme_classic() + 
  geom_point(aes(colour=padj < 0.05), size=1) +
  coord_cartesian(xlim = xlim) +
  geom_text_repel(data = filtTab %>%
                    dplyr::filter(padj < 0.05,
                                  abs(log2FoldChange) > 0.5),
                  aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol),
                  size = 3,
                  max.overlaps = 20,
                  force = 2) +
  scale_color_manual(values = as.character(merck_colors)[c(1,6)]) +
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed",
             color = "grey") +
  geom_vline(xintercept = 0,
             linetype = "dashed",
             color = "grey") +
  labs(x = "log2 fold Change",
       y = "-log10(p adj)",
       title = "Notch On (Jagged1) vs. Control (FC)")
  
```

### Jagged1 vs DAPT

```{r 11a-DE-Jagged1-DAPT-vulcano, fig.width=4, fig.height=4, dev = "png"}

res_Jagged1_DAPT <- results(ddsSE, contrast=c("condition_treatment","Jagged1","DAPT"))

hist(res_Jagged1_DAPT$pvalue)

#We can order our results table by the smallest p value:

res_Jagged1_DAPTOrdered <- res_Jagged1_DAPT[order(res_Jagged1_DAPT$padj),]
#We can summarize some basic tallies using the summary function.

summary(res_Jagged1_DAPT, alpha = 0.05)

plotMA(res_Jagged1_DAPT, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. DAPT\nlog2 fold changes")

printDF_Jagged1_DAPT <- res_Jagged1_DAPT %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(se_counts_filtered) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

  DT::datatable(printDF_Jagged1_DAPT %>% 
          mutate_if(is.numeric, ~signif(., 2)), 
            class = 'cell-border stripe', 
            extensions = 'Buttons',
            rownames = TRUE,
            options = list(
              scrollX='600px',
              dom = 'Bfrtip',
              buttons = c('excel'),
              headerCallback = DT::JS(
                "function(thead) {",
                "  $(thead).css('font-size', '0.7em');",
                "}"
                )
              )
            ) %>%
    DT::formatStyle(columns = c(1:ncol(printDF_Jagged1_DAPT)), fontSize = '70%')
```

#### Vulcaono Plot

```{r 11b-DE-Jagged1-DAPT-vulcano, fig.width=4, fig.height=4}

# resultsNames(ddsSE)
# For visualization of Vulcano Plots, log fold changes are shrunken by "ashr"

resLFC <- lfcShrink(ddsSE, contrast=c("condition_treatment","Jagged1","DAPT"), type="ashr")

# summary(resLFC, alpha = 0.05)

plotMA(resLFC, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. DAPT\nAshr shrinked log2 fold changes")

# first remove the filtered genes (FDR=NA) and create a -log10(FDR) column
filtTab <- resLFC %>%
  as.data.frame() %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(se_counts_filtered) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol),
                   by = "ensembl") %>%
    mutate(`-log10(padj)` = -log10(padj)) %>%
  dplyr::filter(!is.na(`-log10(padj)`))


xlim <- range(filtTab$log2FoldChange)
xlim[1] <- ifelse(xlim[1] > -2, -2, xlim[1])
xlim[2] <- ifelse(xlim[2] < 2, 2, xlim[2])
xlim[1] <- ifelse(abs(xlim[2]) > abs(xlim[1]), -xlim[2], xlim[1])
xlim[2] <- ifelse(abs(xlim[1]) > abs(xlim[2]), -xlim[1], xlim[2])

ggplot(filtTab %>%
         dplyr::filter(!is.na(padj)), 
       aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol)) + 
  theme_classic() + 
  geom_point(aes(colour=padj < 0.05), size=1) +
  coord_cartesian(xlim = xlim) +
  geom_text_repel(data = filtTab %>%
                    dplyr::filter(padj < 0.05,
                                  abs(log2FoldChange) > 0.5),
                  aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol),
                  size = 3,
                  max.overlaps = 20,
                  force = 2) +
  scale_color_manual(values = as.character(merck_colors)[c(1,6)]) +
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed",
             color = "grey") +
  geom_vline(xintercept = 0,
             linetype = "dashed",
             color = "grey") +
  labs(x = "log2 fold Change",
       y = "-log10(p adj)",
       title = "Notch On (Jagged1) vs. Notch Off (DAPT)")
  
```

## Overlap of differentially expressed genes between DAPT vs. FC and Jagged1 vs. FC

### Upregulated in DAPT and Jagged1

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}
overlap_both_up <- calculate.overlap(x = list(
    "Upregulated Genes Jagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Genes DAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated Genes\nJagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Genes\nDAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in DAPT vs. FC and Jagged1 vs. FC\ntreated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.025, 0.025),
  cat.pos = c(-5, 5),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)
```

### Downregulated in DAPT and Jagged1

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}
overlap_both_down <- calculate.overlap(x = list(
    "Downregulated Genes Jagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Genes DAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated Genes\nJagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Genes DAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in DAPT vs. FC and Jagged1 vs. FC\ntreated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.025, 0.025),
  cat.pos = c(-160, -25),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```

### Upregulated in DAPT vs. FC and Downregulated in Jagged1 vs. FC

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}
overlap_DAPT_up_Jagged1_down <- calculate.overlap(x = list(
    "Downregulated Genes Jagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Genes DAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

print("Overlap:")
print(overlap_DAPT_up_Jagged1_down$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated Genes\nJagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Genes DAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in DAPT vs. FC\nand Downregulated Genes in Jagged1 vs. FC\ntreated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.055, 0.055),
  cat.pos = c(25, 5),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```

### Downregulated in DAPT vs. FC and Upregulated in Jagged1 vs. FC

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}
overlap_DAPT_down_Jagged1_up <- calculate.overlap(x = list(
  "Upregulated Genes Jagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Genes DAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

print("Overlap:")
print(overlap_DAPT_down_Jagged1_up$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated Genes Jagged1 vs. FC" = printDF_Jagged1 %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Genes DAPT vs. FC" = printDF_DAPT %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in DAPT vs. FC\nand Upregulated Genes in Jagged1 vs. FC\ntreated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.015, 0.015),
  cat.pos = c(35, 5),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```

```{r 13-save}

save(ddsSE,
     meta.all_summary,
     qnt.mat_tpm_agg_log2,
     notch_CCLE_mutations_csv_21Q1_long,
     ddsSE_preparation,
     res_Jagged1_DAPT,
     res_Jagged1,
     res_DAPT,
     file = "../RData/Part1a.RData")
```

# Discussion of the Results and Recommendations

Differentially expression analysis of the training data set (both 8h and 72h) of the six basal-like cell lines resulted in lists of differentially expressed genes. In a next step, differentially expression analysis will be performed for the 8h and 72h timepoint seperately.    

# References {.unlisted .unnumbered}

<div id="refs"></div>

# Appendix {.unlisted .unnumbered}
  
```{r SessionInfo}
pander::pander(sessionInfo())
```