---
title: "Project: Identification of a Notch transcriptomic signature for breast cancer"
subtitle: "Part 8 - CSL binding motif"
author: "Computational Oncology: Felix Geist"
date: "`r Sys.Date()`"
output: 
  html_document:
    theme: flatly
    highlight: tango
    fig_caption: TRUE
    number_sections: FALSE
    code_folding: hide
    toc: TRUE
    toc_depth: 1
    toc_float:
      collapsed: FALSE
      smooth_scroll: FALSE
bibliography: ["`r here::here('src', 'part8.bib')`","`r here::here('src', 'part8_references.bib')`"]
link-citations: true
biblio-style: "apalike"
---

```{r 00_setup, include=FALSE}

knitr::opts_chunk$set(
	fig.height = 4,                  # 4 inch high, 6 inch wide
	fig.width = 6,
	dpi = 300,                       
	dev = c("svg"),                  
	fig.align = "center",
	# fig.path = "../Figures/Part8/",           
	message = FALSE,
	warning = FALSE,
	echo = TRUE,                     
	cache = FALSE                    
)
```

```{css, echo=FALSE}

text {
  font-family: sans-serif;
}

h1, .h1, h2, .h2, h3, .h3 {
    margin-top: 0.5em;
}

div.box { 
  background-color:#EEEEEE; 
  border: 2px solid #0F69AF;
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

div.obs { 
  background-color:white; 
  border: 2px solid #0F69AF; 
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

body, td {
   font-size: 14px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 14px
}

img {
    max-width: 80%;
}
```

```{r 01-packages, echo=TRUE, message=FALSE, warning=FALSE, results='hide'}

library(tidyverse)

library(readxl)
library(xlsx)
library(knitr)
library(kableExtra)

library(ggplot2)
library(ggpubr)
library(ggrepel)

library(circlize)
library(ggthemes)
library(ComplexHeatmap)
library(rmarkdown)
library(DT)

library(rstatix)

library(viridis)

### Motif analysis:

library(MotifDb)
library(S4Vectors)
library(seqLogo)
library(motifStack)
library(grImport2)
library(grImport)
library(Biostrings)
library(GenomicFeatures)
library(org.Hs.eg.db)
library(BSgenome.Hsapiens.UCSC.hg38)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(clusterProfiler)

# Merck X-Omics Platform:

library(ProfilerAPI2)
# library(xopdata)
# 
# getsrc <- xopdata::get_src()

# Citations:
knitr::write_bib(file = '../src/part8.bib')

# color schema

merck_colors <- ggthemes::tableau_color_pal("Color Blind")(10)

merck_colors_ext <- colorRamp2(seq(from = 1, to = 40, by = 4), merck_colors[1:10])

subtype_colors <- merck_colors[c(5,9,6)]
names(subtype_colors) <- c("basal", "her2", "luma")

all_subtype_colors <- c(subtype_colors, c("lumb" = as.character(merck_colors[4])))

time_colors <- merck_colors[c(10,7)]
names(time_colors) <- c("8h", "72h")

treatment_colors <- merck_colors[c(1,2,3,4)]
names(treatment_colors) <- c("Notch on", "Notch off", "ground state","Notch on/off")

batch_colors <- merck_colors[c(5,9)]
names(batch_colors) <- c("Validation", "Training")

col_heatmap <- c(merck_colors[1], "white", merck_colors[6])
col_heatmap_not0 <- c(merck_colors[1], merck_colors[3], merck_colors[6])

theme_set(theme_bw())

# heatmap colors:

options(knitr.kable.NA = '')   # This replaces NA in knitr tables to an empty string

```

**Research Questions**

<div class = "box">

We want to identify general applicable signatures for the activation and inactivation of the NOTCH pathway. Therefore, we treated triple negative breast cancer cell lines either with the gamma-secretase inhibitor DAPT or cultivated the cells on immobilized NOTCH ligand Jagged1. Subsequently, the cells were harvested after 8h or 72h of incubation time and their transcriptome analyzed by RNASeq.  

</div>
  
**Methods**

With the help of the `SwissRegulon` [@SwissRegulon] and `JASPAR2018` [@JASPAR2018] (stored in `MotifDB` [@R-MotifDb]), the binding sequence of CSL (RBPJ) has been identified and within the promotor region of the signature genes searched for (+/- 3000 bp).  

**Conclusion**  

<div class = "box">

All signature genes contain at least one binding sequence for CSL in any of their transcript variants, with all canonical sequences covered.  

</div>

# Methods and Data Sources

## Methods

**Platform and Report**  

The analysis was performed using R `r paste(R.Version()[c("major", "minor")], collapse = ".")` [@R-base] with the extension of the `tidyverse` [@R-tidyverse; @tidyverse2019] (dplyr, forcats, ggplot2, purrr, readr, stringr, tibble, tidyr).   
The report was generated using `Rmarkdown` [@R-rmarkdown; @rmarkdown2018; @rmarkdown2020] and `knitr` [@R-knitr; @knitr2014; @knitr2015].  

**Transcription Factor Binding**  
Motifs were identified by using `MotifDB` [@R-MotifDb] and subsequently `SwissRegulon` [@SwissRegulon] and `JASPAR2018` [@JASPAR2018]. 
Transcrip variants and promoter sequences of the signature genes were annotated using `TxDb.Hsapiens.UCSC.hg38.knownGene` [@R-TxDb.Hsapiens.UCSC.hg38.knownGene] and `GenomicFeatures` [@R-GenomicFeatures; @GenomicFeatures2013]. The CSL motif within the promotor sequences were identified using a Position Weight Matrix (PWM) with the help of `Biostrings` [@R-Biostrings]. Canonical sequences were annotated by using the [UCSC](http://genome.ucsc.edu/cgi-bin/hgTables?hgsid=1412888849_rKl43K1tPzDSPfIqfyeoGkuta8Vr&clade=mammal&org=Human&db=hg38&hgta_group=genes&hgta_track=knownGene&hgta_table=knownCanonical&hgta_regionType=genome&position=chrX%3A15%2C560%2C138-15%2C602%2C945&hgta_outputType=primaryTable&hgta_outFileName=).  

**Visualization**  
Plots are drawn by `ggplot2` and `motifStack` [@R-motifStack; @motifStack2018]
Tables are drawn, using the `knitr` package, together with `kableExtra` styling [@R-kableExtra]. Interactive Tables are integrated using the `DT` package [@R-DT].  

# Load Data

```{r}
base::load("../RData/Part3.RData")

```

## CSL sequence: Transcription factor binding 

Are the genes that are regulated by the screens bound by the DNA binding element of the NOTCH pathway (CSL / RBPJ)?  

### CSL Motif

CSL motif from:  

* SwissRegulon [@SwissRegulon]
* JASPAR2018 [@JASPAR2018]

```{r 02-CSL-binding-motif,fig.height=6, fig.width=6}

RBPJ_binding <- query(MotifDb, "RBPJ", andStrings = "Hsapiens")

pfm.RBPJ.jaspar2018 <- new("pfm", mat=query(MotifDb, "RBPJ", andStrings = "Hsapiens")[[2]], 
                        name=names(RBPJ_binding)[2])
pfm.RBPJ.SwissRegulon <- new("pfm", mat=query(MotifDb, "RBPJ", andStrings = "Hsapiens")[[1]], 
                        name=names(RBPJ_binding)[1])

motifStack::plotMotifLogoStack(motifStack::DNAmotifAlignment(c(pfm.RBPJ.jaspar2018, pfm.RBPJ.SwissRegulon)))

```

from the vignette of the "generegulation" package (https://www.bioconductor.org/packages/release/workflows/vignettes/generegulation/inst/doc/generegulation.html):

To match motifs in a promoter, these steps are required:

* Retrieve the binding motif (the position frequency matrix, or PFM) of a given transcription factor
* Retrieve the promoter regions for a set of candidate targets
* Identify the sequence matches of the binding motif in the the genesâ€™ promoter regions

### Test known NOTCH regulated genes: Consensus sequence within 3000 bp up- and downstream from TSS

Used Motif: SwissRegulon, 85% sequence homology  

```{r 03-known-csl-targets,fig.height=6, fig.width=6}

upstream_TSS <- 3000
downstream_TSS <- 500

pfm.RBPJ.jaspar2018 <- query(MotifDb, "RBPJ", andStrings = "Hsapiens")[[1]]
pcm.RBPJ.jaspar2018 <- round(100 * pfm.RBPJ.jaspar2018)

pfm.RBPJ.SwissRegulon <- query(MotifDb, "RBPJ", andStrings = "Hsapiens")[[2]]
pcm.RBPJ.SwissRegulon <- round(100 * pfm.RBPJ.SwissRegulon)

# Identical sequences in swiss and jaspar 2018 -> take swiss regulon

genes <- c("HEY1", "HES1", "NRARP", "HES5")
eid <- AnnotationDbi::select(org.Hs.eg.db, genes, "ENTREZID", "SYMBOL")[["ENTREZID"]]

grl <- transcriptsBy(TxDb.Hsapiens.UCSC.hg38.knownGene, by="gene")[eid]

promoter.seqs <- getPromoterSeq(grl, Hsapiens, upstream=upstream_TSS,
                                downstream=downstream_TSS)

promoter.seqs <- unlist(promoter.seqs)

hits <- matchPWM(pcm.RBPJ.jaspar2018, promoter.seqs[[1]], "85%", with.score = TRUE)
htits_rv <- matchPWM(reverseComplement(pcm.RBPJ.jaspar2018), promoter.seqs[[1]])

RBPJ.SwissRegulon.hits  <- sapply(promoter.seqs, function(pseq) 
                            matchPWM(pcm.RBPJ.SwissRegulon, pseq, min.score="85%"))

RBPJ.SwissRegulon  <- sapply(RBPJ.SwissRegulon.hits,  length)
names(RBPJ.SwissRegulon) <- stringr::str_remove(names(RBPJ.SwissRegulon),"[.][0-9]{1,}$")
names(RBPJ.SwissRegulon) <- AnnotationDbi::select(org.Hs.eg.db, names(RBPJ.SwissRegulon), "SYMBOL", "ENTREZID")[["SYMBOL"]]

tbl.RBPJ     <- data.frame(gene=names(RBPJ.SwissRegulon), tx_name = promoter.seqs@elementMetadata$tx_name, RBPJ.SwissRegulon)

tbl.RBPJ <- aggregate(tx_name ~ gene + RBPJ.SwissRegulon, 
                                              data = tbl.RBPJ, 
                                              FUN = function(x) paste(unique(x), collapse="; "))


kable(tbl.RBPJ)

```

# RBPJ binding sequence in promotor of signature genes

```{r 04-8h-upregulated-genes-csl-binding,fig.height=6, fig.width=6}

eg = bind_rows(map(gs.list.opt_notch, function(x) bitr(x, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Hs.eg.db")), .id = "Signature")

information_TSS <- AnnotationDbi::select(TxDb.Hsapiens.UCSC.hg38.knownGene, keys = unique(eg$ENTREZID), columns=c("TXNAME","TXCHROM","TXSTART","TXEND","TXSTRAND"), keytype="GENEID")

information_TSS <- information_TSS %>%
  na.omit()

grl <- transcriptsBy(TxDb.Hsapiens.UCSC.hg38.knownGene, by="gene")[unique(information_TSS$GENEID)]

gene_id_list <- information_TSS %>% 
  dplyr::select(GENEID, TXNAME) %>% 
  dplyr::distinct()

information_TSS <- information_TSS %>%
  dplyr::select(-GENEID)  

promoter.seqs <- getPromoterSeq(grl, Hsapiens, upstream=upstream_TSS,
                                downstream=downstream_TSS)

promoter.seqs <- unlist(promoter.seqs)

RBPJ.SwissRegulon.hits  <- sapply(promoter.seqs, function(pseq) 
                            matchPWM(pcm.RBPJ.SwissRegulon, pseq, min.score="85%"))

RBPJ.SwissRegulon  <- sapply(RBPJ.SwissRegulon.hits,  length)
names(RBPJ.SwissRegulon) <- stringr::str_remove(names(RBPJ.SwissRegulon),"[.][0-9]{1,}$")
names(RBPJ.SwissRegulon) <- AnnotationDbi::select(org.Hs.eg.db, names(RBPJ.SwissRegulon), "SYMBOL", "ENTREZID")[["SYMBOL"]]

tbl.RBPJ <- data.frame(gene=names(RBPJ.SwissRegulon), 
                       tx_name = promoter.seqs@elementMetadata$tx_name, 
                       directionality = information_TSS$TXSTRAND,
                       n_RBPJ_in_tx = RBPJ.SwissRegulon) %>% 
  distinct()

# Annotate transcription start site

tbl.RBPJ <- tbl.RBPJ %>%
  dplyr::left_join(information_TSS,
                   by = c("tx_name" = "TXNAME"))

# annotate distance from TSS + sequence:

tbl.RBPJ_TSS_df <- list()

for(i in 1:length(names(RBPJ.SwissRegulon.hits))){
  
  TSS_paste <- as.data.frame(RBPJ.SwissRegulon.hits[[i]]@ranges)$start-upstream_TSS
  motif <- as.character(RBPJ.SwissRegulon.hits[[i]])
  
  if(length(TSS_paste)==0) {TSS_paste = NA}
  if(length(motif)==0) {motif = NA}

  tbl.RBPJ_TSS_df[[i]] <- bind_cols(tbl.RBPJ[i,],
                                    data.frame(Entrez = str_remove(names(RBPJ.SwissRegulon.hits)[i],
                                                                   "\\..+$"),
                                               Distance_to_TSS = TSS_paste,
                                               motif = motif))
}

tbl.RBPJ_TSS_df <- bind_rows(tbl.RBPJ_TSS_df)

# Get Known Canonical sequences from UCSC:
# http://genome.ucsc.edu/cgi-bin/hgTables?hgsid=1412888849_rKl43K1tPzDSPfIqfyeoGkuta8Vr&clade=mammal&org=Human&db=hg38&hgta_group=genes&hgta_track=knownGene&hgta_table=knownCanonical&hgta_regionType=genome&position=chrX%3A15%2C560%2C138-15%2C602%2C945&hgta_outputType=primaryTable&hgta_outFileName=

hgTables_KnownCanonical_csv <- read_delim("../data/hgTables_KnownCanonical.csv.gz", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)

tbl.RBPJ_TSS_df <- tbl.RBPJ_TSS_df %>% 
  dplyr::left_join(hgTables_KnownCanonical_csv %>% 
                     dplyr::select(transcript) %>% 
                     dplyr::mutate(canonical = "canonical"),
                   by = c("tx_name" = "transcript"))

DT::datatable(tbl.RBPJ_TSS_df %>% 
                dplyr::arrange(gene), 
          class = 'cell-border stripe', 
          extensions = 'Buttons',
          rownames = FALSE,
          options = list(
            dom = 'Bfrtip',
            buttons = c('excel'),
            headerCallback = DT::JS(
              "function(thead) {",
              "  $(thead).css('font-size', '0.7em');",
              "}"
              )
            )
          ) %>%
  DT::formatStyle(columns = c(1:ncol(tbl.RBPJ_TSS_df)),fontSize = '70%')


```

# RBPJ binding sequence in promotor of signature genes (only canonical sequences)

If two canonical sequences are annotated, use the longer version.  

```{r}

tbl.RBPJ_TSS_df_can <- tbl.RBPJ_TSS_df %>% 
  dplyr::filter(!is.na(canonical))

tbl.RBPJ_TSS_df_can_TSS <- tbl.RBPJ_TSS_df_can %>% 
  dplyr::select(-Distance_to_TSS, -motif, -canonical) %>% 
  dplyr::distinct() %>% 
  dplyr::left_join(tbl.RBPJ_TSS_df_can %>% 
                     dplyr::select(tx_name, Distance_to_TSS, motif) %>% 
                     dplyr::group_by(tx_name) %>% 
                     dplyr::summarise(Distance_to_TSS = paste(Distance_to_TSS, collapse = "; "),
                                      motif = paste(motif, collapse = "; ")),
                   by = "tx_name") %>% 
  dplyr::arrange(gene)

filtered_transcripts <- tbl.RBPJ_TSS_df_can_TSS %>% 
                                  dplyr::filter(gene %in% !!tbl.RBPJ_TSS_df_can_TSS$gene[duplicated(tbl.RBPJ_TSS_df_can_TSS$gene)]) %>% 
                                  dplyr::mutate(length = abs(TXEND-TXSTART)) %>% 
                                  dplyr::group_by(gene) %>% 
                                  dplyr::arrange(desc(length)) %>% 
                                  dplyr::filter(!row_number() %in% 1) %>% 
                                  dplyr::ungroup() %>% 
                                  dplyr::pull(tx_name)

tbl.RBPJ_TSS_df_can_TSS <- tbl.RBPJ_TSS_df_can_TSS %>% 
                                  dplyr::filter(!tx_name %in% !!filtered_transcripts)


DT::datatable(tbl.RBPJ_TSS_df_can_TSS, 
          class = 'cell-border stripe', 
          extensions = 'Buttons',
          rownames = FALSE,
          options = list(
            dom = 'Bfrtip',
            buttons = c('excel'),
            headerCallback = DT::JS(
              "function(thead) {",
              "  $(thead).css('font-size', '0.7em');",
              "}"
              )
            )
          ) %>%
  DT::formatStyle(columns = c(1:ncol(tbl.RBPJ_TSS_df_can_TSS)),fontSize = '70%')

```


# References {.unlisted .unnumbered}

<div id="refs"></div>

# Appendix {.unlisted .unnumbered}
  
```{r SessionInfo}
pander::pander(sessionInfo())

```
