---
title: "Project: Identification of a Notch transcriptomic signature for breast cancer"
subtitle: "Part 1c - Training Data: Basal-like treated BRCA Cell Lines, 72h treatment"
author: "Computational Oncology: Felix Geist"
date: "`r Sys.Date()`"
output: 
  html_document:
    theme: flatly
    highlight: tango
    fig_caption: TRUE
    number_sections: FALSE
    code_folding: hide
    toc: TRUE
    toc_depth: 1
    toc_float:
      collapsed: FALSE
      smooth_scroll: FALSE
bibliography: "`r here::here('src', 'part1c.bib')`"
link-citations: true
biblio-style: "apalike"
---

```{r 00_setup, include=FALSE}

### --- MOST IMPORTANT: Always comment your code, so people can follow the structure

# First two chunks are to set the settings needed for the template
# numbering and naming of the chunks will directly name the files of the figures that are produced

knitr::opts_chunk$set(
	fig.height = 4,                  # 4 inch high, 6 inch wide
	fig.width = 6,
	dpi = 300,                       # good resolution for printing/retina displays, increases file size massively
	dev = c("svg"),                  # change that to png if majority of plots are heatmaps or scatter plots. 
	                                 # For single data heavy figures, manually set the chunk options
	fig.align = "center",
	# fig.path = "../Figures/Part1c/",           # if you want all the figures directly written into a seperate folder
	message = FALSE,
	warning = FALSE,
	echo = TRUE,                     # if set to false, the code will be omitted
	cache = FALSE                    # could be useful during the development of a new project
	                                 # don't forget to exclude the cache folder in the .gitignore
)
```

```{css, echo=FALSE}

text {
  font-family: sans-serif;
}

h1, .h1, h2, .h2, h3, .h3 {
    margin-top: 0.5em;
}

div.box { 
  background-color:#EEEEEE; 
  border: 2px solid #0F69AF;
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

div.obs { 
  background-color:white; 
  border: 2px solid #0F69AF; 
  border-radius: 10px; 
  padding: 1em 1em 1em 4em;
  margin: 1em 0 1em 0;
}

body, td {
   font-size: 14px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 14px
}

img {
    max-width: 80%;
}
```

```{r 01-packages, echo=TRUE, results = 'hide'}

library(tidyverse)
library(here)

library(xlsx)

library(ggplot2)
library(ggpubr)
library(ggrepel)
library(circlize)
library(ComplexHeatmap)
library(ggthemes)

library(DESeq2)
library(vsn)
library(PCAtools)
library(uwot)
library(VennDiagram)

library(rmarkdown) 
library(knitr) 
library(kableExtra)

library(rstatix)

library(DESeq2)

library(BiocParallel)
# ncores = detectCores()-2
# register(SnowParam(ncores))

knitr::write_bib(file = '../src/part1c.bib')

# color schema

merck_colors <- ggthemes::tableau_color_pal("Color Blind")(10)

merck_colors_ext <- colorRamp2(seq(from = 1, to = 40, by = 4), merck_colors[1:10])

subtype_colors <- merck_colors[c(5,9,6)]
names(subtype_colors) <- c("basal", "her2", "luma")

time_colors <- merck_colors[c(10,7)]
names(time_colors) <- c("8h", "72h")

treatment_colors <- merck_colors[c(1,2,3,4)]
names(treatment_colors) <- c("Notch on", "Notch off", "ground state","Notch on/off")

col_heatmap <- c(merck_colors[1], "white", merck_colors[6])
col_heatmap_not0 <- c(merck_colors[1], merck_colors[3], merck_colors[6])

theme_set(theme_bw())

options(knitr.kable.NA = '')   # This replaces NA in knitr tables to an empty string

```


**Introduction**

We want to identify robust NOTCH signatures for the activation and inactivation of the NOTCH pathway. Therefore, the group of Urban Lendahl at the Karolinska Institutet, Stockholm, Schweden and the group of Dirk Wienke at the Translational Innovation Platform Oncology and Immuno-Oncology at Merck KGaA, Darmstadt, Germany teamed up.  

<div class = "box">
**Research Questions**

*Treatments:*  

We treated six basal-like breast cancer cell lines either with the gamma-secretase inhibitor DAPT or cultivated the cells on immobilized NOTCH ligand Jagged1. Subsequently, the cells were harvested after 8h or 72h of incubation time and their transcriptome analyzed by RNASeq.  

Each condition was treated for both 8h and for 72h:  

* normal (no treatment): "FC"
* NOTCH ON (cultured on immobilised Jagged1 ligand): "Jagged1"
* NOTCH OFF (blocked with GSI): "DAPT"

Of special interest is the comparison: Jagged1 (NOTCH high) vs. DAPT (NOTCH low)  

**Specific goals:**  

Identify a Notch signature:  

* that defines active and hyperactive notch
* that may identify responsive cell lines / patients

</div>
  
**Methods**

Here, differential expression analysis has been performed only for the 72h time point.  

**Conclusion**  

<div class = "box">

Differentially expression analysis of the training data set (72h only) of the six basal-like cell lines resulted in lists of differentially expressed genes.  

</div>

# Methods and Data Sources

## Methods

**Platform and Report**  
The analysis was performed using R `r paste(R.Version()[c("major", "minor")], collapse = ".")` [@R-base] with the extension of the `tidyverse` [@R-tidyverse; @tidyverse2019] (dplyr, forcats, ggplot2, purrr, readr, stringr, tibble, tidyr).   
The report was generated using `Rmarkdown` [@R-rmarkdown; @rmarkdown2018; @rmarkdown2020] and `knitr` [@R-knitr; @knitr2014; @knitr2015].
Communication with excel files was performed using the `xlsx` package [@R-xlsx]. 

**Expression Profiling**  
The RNASeq data set and data from public ressources was analyzed using `DESeq2` [@R-DESeq2]. Dimensionality of the data set was reduced by `PCAtools` [@R-PCAtools] and UMAPs generated by `uwot` [@R-uwot]. 

**Visualization**  
Plots are drawn by `ggplot2`, `ggthemes` [@R-ggthemes], `ggpubr` [@R-ggpubr] and `ggrepel` [@R-ggrepel]. Tables are drawn, using the `knitr` package, together with `kableExtra` styling [@R-kableExtra]. 

# Dataimport

We import the data of the previous analysis (04-Notch-Part1.html).  

```{r 02-load-data}

base::load("../RData/Part1a.RData")

```

# DESEQ2

We will make use of the `DESeq2` pipeline to identify differentially expressed genes.  

```{r 03-DESEQ2-72h}

# filter 72h

ddsSE_72h_preparation_72h <- ddsSE_preparation[,ddsSE_preparation$condition_time == "72h"]
design(ddsSE_72h_preparation_72h) <- ~ condition_lib_prep + condition_cell_line + condition_treatment

ddsSE_72h <- DESeq(ddsSE_72h_preparation_72h)

```

# QC

For unsupervised clustering and PCA analyses, data will be variance stabilized.   

```{r 04a-DESEQ2-QC-meanSD-72h, include=FALSE}

# this gives log2(n + 1)
ntd <- normTransform(ddsSE_72h)
p <- meanSdPlot(assay(ddsSE_72h), 
                plot = FALSE)
```

```{r 04b-DESEQ2-QC-meanSD-72h}
p$gg +
  ggtitle("Diagnostic Plot: Standard deviations vs row means [log2(n+1)] (72h)")
```

```{r 04c-DESEQ2-QC-meanSD-72h, include=FALSE}
# variance stabilized
vsd <- vst(ddsSE_72h, blind=FALSE)
p <- meanSdPlot(assay(vsd), 
                plot = FALSE)
```

```{r 04d-DESEQ2-QC-meanSD-72h}
p$gg +
  ggtitle("Diagnostic Plot: Standard deviations vs row means [variance stabilized (vst)] (72h)")

```

## Diagnostic Plots

* sample quality
* PCA  
* Screeplots  
* Eigenplots  
* Pairplots

```{r}

meta.all_summary_72h <- meta.all_summary %>% 
  dplyr::filter(condition_time == "72h")

### --- Sample Quality

ggplot(meta.all_summary_72h,
       aes(x=condition_cell_line,
           y=meta_rin)) +
  geom_point() +
  ggtitle("RIN values by cell line (72h only)") +
  labs(y = "RIN values",
       x = "Cell Lines")

ggplot(meta.all_summary_72h,
       aes(x=condition_cell_line,
           y=meta_lib_conc)) +
  geom_point() +
  ggtitle("Library Concentrations by cell line (72h only)") +
  labs(y = "library concentration [nM]",
       x = "Cell Lines")


ggplot(meta.all_summary_72h,
       aes(x=condition_cell_line,
           y=meta_lib_size_bp)) +
  geom_point() +
  ggtitle("Average library size by cell line (72h only)") +
  labs(y = "average library size [bp]",
       x = "Cell Lines")
```

```{r 5a-PCA-correlations-72h, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 12, fig.height=4}

############################# Principal Component analysis ############################# 

qnt.met_PCA <- colData(vsd) %>%
  as.data.frame() %>%
  dplyr::select(-c("files", "names", "sizeFactor"))

qnt.met_PCA$condition_treatment <- factor(qnt.met_PCA$condition_treatment, levels = c("DAPT", "FC", "Jagged1"),
                                      labels = c("Notch off", "ground state", "Notch on"))

qnt.met_PCA$Treatment <- paste(qnt.met_PCA$condition_treatment, "-",qnt.met_PCA$condition_time)

qnt.met_PCA$condition_time <- droplevels(qnt.met_PCA$condition_time)

qnt.met_PCA$sample_name <- rownames(qnt.met_PCA)

p <- pca(assay(vsd), metadata = qnt.met_PCA, removeVar = 0.1)

horn <- parallelPCA(assay(vsd))
#horn$n

elbow <- findElbowPoint(p$variance)
#elbow

eigencorplot(p,
             components = getComponents(p, 1:20),
             metavars = c('condition_lib_prep','condition_cell_line','condition_treatment', 'condition_replicate',
                          'meta_rin', "meta_lib_conc", "meta_lib_size_bp"),
             col = c('white', merck_colors[5], merck_colors[1]),
             cexCorval = 0.7,
             colCorval = 'black',
             fontCorval = 2,
             posLab = 'bottomleft',
             posColKey = 'right',
             rotLabX = 45,
             cexLabColKey = 1.5,
             scale = TRUE,
             main = 'PC1-20 correlations (72h)',
             colFrame = 'white',
             plotRsquared = TRUE)


```

```{r 05b-PCA-screeplot-72h, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 4, fig.height=3.3, dev='png'}

screeplot(p,
          components = getComponents(p, 1:20),
          vline = c(horn$n, elbow),
          colBar = merck_colors[1],
          colCumulativeSumLine = merck_colors[6]) +
  theme_classic() + 
  geom_text(aes(horn$n - 1.5, 30, label = "Horn's", vjust = -1)) +
  geom_text(aes(elbow + 1.5, 30, label = "Elbow", vjust = -1)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(title = "Screeplot - 72h")

```

### PCA

```{r 05c-PCA-72h, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 6, fig.height=5}

biplot(p,
       title = "PC1 vs PC2",
       subtitle = "Cell Line and treatment condition (72h)",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))

biplot(p,
       title = "PC1 vs PC2",
       subtitle = "Library Prep Batch and treatment condition (72h)",
       lab = NA,
       colby = 'condition_lib_prep',
       shape = "Treatment",
       colkey = as.character(merck_colors[c(8,4)]),
       hline = 0, vline = 0,
       legendPosition = 'right') +
  scale_shape_manual(values = c(0,15,1,16,2,17))


biplot(p,
       x = "PC5",
       y = "PC8",
       title = "PC5 vs PC8",
       subtitle = "Treatment and Cell Lines (72h)",
       lab = NA,
       colby = 'condition_cell_line',
       shape = "Treatment",
       colkey = as.character(merck_colors),
       hline = 0, vline = 0,
       legendPosition = 'right')  +
  scale_shape_manual(values = c(0,15,1,16,2,17))

```

### UMAP

```{r 05d-Umap-72h, echo=TRUE, message=FALSE, warning=FALSE, fig.width= 6, fig.height=5}


z_umap <- umap(scale(assay(vsd)) %>% na.omit() %>% t, pca = 50)

z_umap_df <- data.frame(umap1 = z_umap[,1],
           umap2 = z_umap[,2],
           qnt.met_PCA,
           stringsAsFactors = FALSE)

rownames(z_umap_df) <- rownames(qnt.met_PCA)

ggscatter(z_umap_df,
          x = "umap1",
          y = "umap2",
          #label = "label",
          color = "condition_cell_line",
          shape = "Treatment",
          #alpha = "breast_alpha",
          size = 2,
          legend = "right",
          palette = as.character(merck_colors),
          title = "UMAP Dimensionality Reduction (72h)") + 
  guides(col = guide_legend(nrow = 7)) +
  scale_shape_manual(values = c(0,15,1,16,2,17))

ggscatter(z_umap_df,
          x = "umap1",
          y = "umap2",
          #label = "condition_cell_line",
          #repel = TRUE,
          color = "condition_cell_line",
          shape = "Treatment",
          #alpha = "breast_alpha",
          size = 2,
          legend = "right",
          palette = as.character(merck_colors),
          title = "UMAP Dimensionality Reduction (72h)") +
  geom_text_repel(data = z_umap_df %>% group_by(condition_cell_line) %>% 
               summarise(umap1 = mean(umap1), 
                         umap2 = mean(umap2)),
             aes(x = umap1,
                 y = umap2,
                 label = condition_cell_line,
                 color = condition_cell_line)) + 
  guides(col = guide_legend(nrow = 7)) +
  scale_shape_manual(values = c(0,15,1,16,2,17))

z_tumap <- tumap(scale(assay(vsd)) %>% na.omit() %>% t, pca = 50)

z_tumap_df <- data.frame(tumap1 = z_tumap[,1],
           tumap2 = z_tumap[,2],
           qnt.met_PCA,
           stringsAsFactors = FALSE)

rownames(z_tumap_df) <- rownames(qnt.met_PCA)

ggscatter(z_tumap_df,
          x = "tumap1",
          y = "tumap2",
          #label = "condition_cell_line",
          #repel = TRUE,
          color = "condition_cell_line",
          shape = "Treatment",
          #alpha = "breast_alpha",
          size = 2,
          legend = "right",
          palette = as.character(merck_colors),
          title = "t-UMAP Dimensionality Reduction (72h)") +
  geom_text_repel(data = z_tumap_df %>% group_by(condition_cell_line) %>% 
               summarise(tumap1 = mean(tumap1), 
                         tumap2 = mean(tumap2)),
             aes(x = tumap1,
                 y = tumap2,
                 label = condition_cell_line,
                 color = condition_cell_line)) + 
  guides(col = guide_legend(nrow = 7)) +
  scale_shape_manual(values = c(0,15,1,16,2,17))
```

## Heatmap of variable genes

```{r 06-hm-top-variant-genes-72h, fig.width=8, fig.height=8, dev = "png"}

ntop <- order(matrixStats::rowVars(assay(vsd)), decreasing = TRUE)[1:1000]
cpm_top <- counts(ddsSE_72h,normalized=TRUE)[ntop,]
colnames(cpm_top) <- qnt.met_PCA$condition_cell_line[colnames(cpm_top) %in% qnt.met_PCA$sample_name]

Subtype <- as.character(qnt.met_PCA$condition_subtype)
Subtype_col <- subtype_colors[levels(qnt.met_PCA$condition_subtype)]

# lib_prep <- as.character(qnt.met_no_ctrl$condition_lib_prep)
# lib_prep_col <- merck_colors[c(4,5)]
# names(lib_prep_col) = unique(lib_prep)

cell_line <- as.character(qnt.met_PCA$condition_cell_line)
cell_line_col <- merck_colors[1:length(unique(cell_line))]
names(cell_line_col) = unique(cell_line)

treatment_time <- as.character(qnt.met_PCA$condition_time)
treatment_time_col <- time_colors[levels(qnt.met_PCA$condition_time)]
#names(treatment_time_col) = unique(treatment_time)

treatment_factor <- qnt.met_PCA$condition_treatment
treatment <- as.character(treatment_factor)
treatment_col <- treatment_colors[levels(treatment_factor)]

col_fun = colorRamp2(c(8, 4, 0), col_heatmap)

col_ha = ComplexHeatmap::columnAnnotation(#Subtype = Subtype,
                                          #`Library Prep` = lib_prep,
                                          `Cell line` = cell_line,
                                          `Time point` = treatment_time,
                                          Treatment = treatment,
                                          col = list(#Subtype = Subtype_col,
                                                     #`Library Prep` = lib_prep_col,
                                                     `Cell line` = cell_line_col,
                                                     `Time point` = treatment_time_col,
                                                     Treatment = treatment_col),
                                          annotation_name_side = "right")

ht <- Heatmap(cpm_top,
        column_title = "Top 1000 variant genes in the breast cancer cell lines (72h)",
        cluster_rows = TRUE,
        cluster_columns = TRUE,
        clustering_distance_rows = "manhattan",
        clustering_distance_columns = "manhattan",
        #clustering_distance_rows = "euclidean",
        #clustering_distance_columns = "euclidean",
        col = col_fun,  
        show_row_names = FALSE,
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 4),
        top_annotation = col_ha,
        heatmap_legend_param = list(
           title = "log2[cpm]"#,
           #title_position = "lefttop-rot"
           )
        )

draw(ht)
# , merge_legend = TRUE, adjust_annotation_extension = TRUE
```

# Diagnositic Plots

## Known Notch targets

Diagnostic plots of the TPM values of known targets of the Notch pathway are used to validate the experimental conditions.  

```{r 07-Diagnostic-Plots-72h, fig.width=8, fig.height=8}

qnt.mat_tpm_agg_log2_72h <- qnt.mat_tpm_agg_log2 %>% 
  dplyr::filter(condition_time == "72h")

############################# Diagnostic Plots ############################# 

diag_plot <- function(gene) {
  plotdata <- qnt.mat_tpm_agg_log2_72h %>%
  filter(symbol == !!gene) %>%
  dplyr::mutate(condition_treatment = dplyr::recode_factor(condition_treatment,
                                                           `DAPT`      = 'Notch off',
                                                           `FC`   = 'ground state',
                                                           `Jagged1`     = 'Notch on'),
                condition_time = factor(condition_time, levels = c("72h")))

  p <- ggplot(plotdata,
         aes(y = quant_log2tpm,
             x = condition_time,
             fill = condition_treatment)) +
    geom_boxplot(position=position_dodge(1), outlier.colour = NA) +
    theme_pubclean() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    geom_point(size = 2, position=position_dodge(1)) +
    ylim(c(0,range(plotdata$quant_log2tpm)[2])) +
    facet_wrap(condition_cell_line~condition_subtype, 
               ncol=3,
               scales = "free")+
    labs(title = paste(gene, "- 72h"),
         x = "Condition",
         y = "log(TPM+1)") +
   # scale_fill_manual(values = time_colors[levels(factor(plotdata$condition_time))])
    scale_fill_manual(values = treatment_colors[levels(factor(plotdata$condition_treatment))])
    
    
  print(p)
}

diag_plot("HEY1")
diag_plot("HES1")
diag_plot("NRARP")
diag_plot("MYC")
diag_plot("IER5")
diag_plot("SAT1")

```

## Notch Expression Heatmap 72h

```{r 08b-notch-expression-hm-72h, fig.height=2.5, fig.width=4}

notch_expression <- qnt.mat_tpm_agg_log2 %>%
  filter(condition_time == "72h") %>%
  filter(condition_treatment == "FC") %>%
  filter(symbol %in% c("NOTCH1","NOTCH2","NOTCH3","NOTCH4")) %>%
  dplyr::select(symbol, quant_log2tpm, condition_cell_line,condition_subtype) %>%
  arrange(symbol, condition_subtype, condition_cell_line)

matrix_notch_expression <- notch_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("symbol")

Subtype <- as.character(notch_expression[,c("condition_cell_line", "condition_subtype")] %>% distinct() %>% pull("condition_subtype"))

Subtype_col <- subtype_colors[unique(Subtype)]

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          col = list(Subtype = Subtype_col), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")
                                            )
)

ht <- ComplexHeatmap::Heatmap(matrix_notch_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                      #  column_split  = notch_expression %>% filter(sample_name == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
)

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legend = TRUE)

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          col = list(Subtype = Subtype_col), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")#,
                                           # Cellline = list(nrow = 4)
                                          )
)

ht <- ComplexHeatmap::Heatmap(matrix_notch_expression,
                        cluster_rows = FALSE,
                        cluster_columns = TRUE,
                        #column_split  = notch_expression %>% filter(gene_symbol == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
                        )

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legend = TRUE)

```

## Endocrine receptor status of cell lines - 72h

```{r 08d-receptors-expression-hm-72h, fig.height=3.5, fig.width=4}
receptor_expression <- qnt.mat_tpm_agg_log2 %>%
  filter(condition_treatment == "FC") %>%
  filter(condition_time == "72h") %>%
  filter(symbol %in% c("ERBB2","ERBB3","ESR1","PGR","AR")) %>%
  dplyr::select(symbol, quant_log2tpm, condition_cell_line, condition_time, condition_subtype) %>%
  arrange(symbol, condition_subtype, condition_cell_line, desc(condition_time))

matrix_receptor_expression <- receptor_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("symbol")

Subtype <- as.character(receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_subtype"))

Subtype_col <- subtype_colors[unique(Subtype)]

Timepoint <- as.character(receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_time"))

Timepoint_col <- time_colors[unique(Timepoint)]


col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          #Cellline = Cellline,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col
                                                     #Cellline = Cellline_col
                                                     ), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")#,
                                            #Cellline = list(nrow = 4,
                                            #                at = matrix_receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% pull("condition_cell_line") %>% unique())
                                            )
)

ht <- ComplexHeatmap::Heatmap(matrix_receptor_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = notch_expression %>% filter(gene_name == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
)

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          #Cellline = Cellline,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col#,
                                                     #Cellline = Cellline_col
                                                     ), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")#,
                                           # Cellline = list(nrow = 4)
                                          )
)

ht <- ComplexHeatmap::Heatmap(matrix_receptor_expression,
                        cluster_rows = FALSE,
                        cluster_columns = TRUE,
                        #column_split  = notch_expression %>% filter(gene_symbol == "NOTCH1") %>% pull(condition_subtype) %>% as.vector,
                        row_title_rot = 90,
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        column_title_gp = gpar(fontsize = 8),
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "log2(TPM+1)", 
                          title_position = "leftcenter-rot")
                        )

draw(ht, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)


```

## Combined plot

```{r 08f-receptors-expression-hm-72h, fig.height=5, fig.width=6}

receptor_expression <- qnt.mat_tpm_agg_log2_72h %>%
  filter(condition_treatment == "FC") %>%
  filter(symbol %in% c("ERBB2","ERBB3","ESR1","PGR","AR")) %>%
  dplyr::select(gene_symbol = symbol, quant_log2tpm, condition_cell_line, condition_subtype, condition_time) %>%
  arrange(gene_symbol, condition_subtype, condition_cell_line, desc(condition_time))

matrix_receptor_expression <- receptor_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("gene_symbol")

notch_expression <- qnt.mat_tpm_agg_log2_72h %>%
  filter(condition_treatment == "FC") %>%
  filter(symbol %in% c("NOTCH1", "NOTCH2", "NOTCH3", "NOTCH4")) %>%
  dplyr::select(gene_symbol = symbol, quant_log2tpm, condition_cell_line, condition_subtype, condition_time) %>%
  arrange(gene_symbol, condition_subtype, condition_cell_line, desc(condition_time))

matrix_notch_expression <- notch_expression %>% 
  dplyr::select(-condition_subtype) %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = quant_log2tpm,
                     values_fn = mean) %>%
  column_to_rownames("gene_symbol") 

notch_CCLE_mutations_use_wide <- notch_expression %>%
  dplyr::select(-gene_symbol, -quant_log2tpm) %>%
  dplyr::distinct() %>%
  left_join(notch_CCLE_mutations_csv_21Q1_long %>%
    dplyr::select(-DepMap_ID),
    by = c("condition_cell_line" = "stripped_cell_line_name")) %>%
  arrange(gene_symbol, condition_subtype, condition_cell_line, desc(condition_time))
 
matrix_notch_mutation <- notch_expression %>% 
  dplyr::select(-condition_subtype, -gene_symbol, -quant_log2tpm) %>%
  dplyr::left_join(notch_CCLE_mutations_use_wide) %>% 
  dplyr::select(-condition_subtype) %>%
  dplyr::distinct() %>%
  tidyr::pivot_wider(names_from = c(condition_cell_line, condition_time),
                     values_from = mutation) %>%
  dplyr::filter(!is.na(gene_symbol)) %>%
  column_to_rownames("gene_symbol")

matrix_notch_mutation <- matrix_notch_mutation[rownames(matrix_notch_mutation) %in% c("NOTCH1", "NOTCH2", "NOTCH3", "NOTCH4"),colnames(matrix_receptor_expression)]

# header

Subtype <- as.character(receptor_expression[,c("condition_cell_line", "condition_subtype", "condition_time")] %>% distinct() %>% pull("condition_subtype"))
Subtype_col <- subtype_colors[unique(Subtype)]

Timepoint <- as.character(receptor_expression[,c("condition_cell_line", "condition_time", "condition_subtype")] %>% distinct() %>% pull("condition_time"))
Timepoint_col <- time_colors[unique(Timepoint)]

col_ha = ComplexHeatmap::columnAnnotation(Subtype = Subtype,
                                          Timepoint = Timepoint,
                                          col = list(Subtype = Subtype_col,
                                                     Timepoint = Timepoint_col), 
                                          #annotation_name_side = "left",
                                          annotation_legend_param = list(
                                            Subtype = list(direction = "horizontal")
                                            )
)

# col_split <- receptor_expression %>% filter(gene_symbol == "ESR1") %>% pull(condition_subtype) %>% as.vector

ht <- ComplexHeatmap::Heatmap(matrix_receptor_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = col_split,
                        row_title = "Expression of\nEndocrine\nReceptors",
                        row_title_rot = 0,
                        row_title_gp = gpar(fontsize = 10),
                        top_annotation = col_ha,
                        show_column_names = TRUE,
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "scaled log2(TPM+1)", 
                          #title_position = "leftcenter-rot",
                          direction = "horizontal")
)

ht2 <- ComplexHeatmap::Heatmap(matrix_notch_expression,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = col_split,
                        row_title = "Expression of\nNotch\nReceptors",
                        row_title_rot = 0,
                        row_title_gp = gpar(fontsize = 10),
                        show_column_names = TRUE,
                        col = col_fun,
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        show_heatmap_legend = FALSE,
                        heatmap_legend_param = list(
                          title = "scaled log2(TPM+1)", 
                          title_position = "leftcenter-rot")
)

mutation_names <- unique(unlist(matrix_notch_mutation))
mutation_names <- mutation_names[!is.na(mutation_names)]

col_mut = c("lightgrey",merck_colors[1:(length(mutation_names)-1)])
names(col_mut) <- mutation_names


ht3 <- ComplexHeatmap::Heatmap(matrix_notch_mutation,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                       # column_split  = col_split,
                        row_title = "Mutations in\nNotch\nReceptors",
                        row_title_rot = 0,
                        row_title_gp = gpar(fontsize = 10),
                        show_column_names = TRUE,
                        col = col_mut,
                        na_col = "white",
                        rect_gp = gpar(col = "white", lwd = 0.5),
                        heatmap_legend_param = list(
                          title = "NOTCH Mutations")
                        
)

draw(ht %v% ht2 %v% ht3, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legends = TRUE)

```

# Differential Gene Expression Analysis

We extract the contrasts of interest from the DESEQ2 object, using a FDR cutoff of 0.05.  

## 72h

### FC vs DAPT - 72h

```{r 09a-DE-FC-DAPT-72h, fig.width=4, fig.height=4, dev = "png"}

res_DAPT_72h <- results(ddsSE_72h, contrast=c("condition_treatment","DAPT","FC"), alpha = 0.05)

hist(res_DAPT$pvalue)

#We can order our results table by the smallest p value:

res_DAPTOrdered <- res_DAPT_72h[order(res_DAPT_72h$padj),]
#We can summarize some basic tallies using the summary function.

summary(res_DAPT_72h, alpha = 0.05)

plotMA(res_DAPT_72h, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot FC vs. DAPT - 72h\nlog2 fold changes")

printDF_DAPT <- res_DAPT_72h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_72h) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol, genename, entrez),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl) %>%
  dplyr::arrange(desc(log2FoldChange))

  DT::datatable(printDF_DAPT %>% 
          mutate_if(is.numeric, ~signif(., 2)), 
            class = 'cell-border stripe', 
            extensions = 'Buttons',
            rownames = TRUE,
            options = list(
              scrollX='600px',
              dom = 'Bfrtip',
              buttons = c('excel'),
              headerCallback = DT::JS(
                "function(thead) {",
                "  $(thead).css('font-size', '0.7em');",
                "}"
                )
              )
            ) %>%
    DT::formatStyle(columns = c(1:ncol(printDF_DAPT)), fontSize = '70%')

```

#### Vulcaono Plot - 72h

```{r 09b-DE-FC-DAPT-vulcano-72h, fig.width=4, fig.height=4}

# resultsNames(ddsSE_72h)

resLFC <- lfcShrink(ddsSE_72h, contrast=c("condition_treatment","DAPT","FC"), type="ashr")

# summary(resLFC, alpha = 0.05)

plotMA(resLFC, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot FC vs. DAPT - 72h\nAshr shrinked log2 fold changes")

# first remove the filtered genes (FDR=NA) and create a -log10(FDR) column
filtTab <- resLFC %>%
  as.data.frame() %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_72h) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol),
                   by = "ensembl") %>%
    mutate(`-log10(padj)` = -log10(padj)) %>%
  dplyr::filter(!is.na(`-log10(padj)`))

xlim <- range(filtTab$log2FoldChange)
xlim[1] <- ifelse(xlim[1] > -2, -2, xlim[1])
xlim[2] <- ifelse(xlim[2] < 2, 2, xlim[2])
xlim[1] <- ifelse(abs(xlim[2]) > abs(xlim[1]), -xlim[2], xlim[1])
xlim[2] <- ifelse(abs(xlim[1]) > abs(xlim[2]), -xlim[1], xlim[2])

ggplot(filtTab %>%
         dplyr::filter(!is.na(padj)), 
       aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol)) + 
  theme_classic() + 
  geom_point(aes(colour=padj < 0.05), size=1) +
  coord_cartesian(xlim = xlim) +
  geom_text_repel(data = filtTab %>%
                    dplyr::filter(padj < 0.05,
                                  abs(log2FoldChange) > 0.5),
                  aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol),
                  size = 3,
                  max.overlaps = 20,
                  force = 2) +
  scale_color_manual(values = as.character(merck_colors)[c(1,6)]) +
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed",
             color = "grey") +
  geom_vline(xintercept = 0,
             linetype = "dashed",
             color = "grey") +
  labs(x = "log2 fold Change",
       y = "-log10(p adj)",
       title = "Notch Off (DAPT) vs. Control (FC) - 72h")

```

### Jagged1 vs FC - 72h

```{r 10a-DE-FC-Jagged1-72h, fig.width=4, fig.height=4, dev = "png"}

res_Jagged1_72h <- results(ddsSE_72h, contrast=c("condition_treatment","Jagged1","FC"))

hist(res_Jagged1_72h$pvalue)

#We can order our results table by the smallest p value:

res_Jagged1Ordered <- res_Jagged1[order(res_Jagged1$padj),]
#We can summarize some basic tallies using the summary function.

summary(res_Jagged1_72h, alpha = 0.05)

plotMA(res_Jagged1_72h, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. FC - 72h\nlog2 fold changes")

printDF_Jagged1 <- res_Jagged1_72h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_72h) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol, genename, entrez),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl) %>%
  dplyr::arrange(desc(log2FoldChange))

  DT::datatable(printDF_Jagged1 %>% 
          mutate_if(is.numeric, ~signif(., 2)), 
            class = 'cell-border stripe', 
            extensions = 'Buttons',
            rownames = TRUE,
            options = list(
              scrollX='600px',
              dom = 'Bfrtip',
              buttons = c('excel'),
              headerCallback = DT::JS(
                "function(thead) {",
                "  $(thead).css('font-size', '0.7em');",
                "}"
                )
              )
            ) %>%
    DT::formatStyle(columns = c(1:ncol(printDF_Jagged1)), fontSize = '70%')
```

#### Vulcaono Plot - 72h

```{r 10b-DE-FC-Jagged1-vulcano-72h, fig.width=4, fig.height=4}
# 
# resultsNames(ddsSE_72h)

resLFC <- lfcShrink(ddsSE_72h, contrast=c("condition_treatment","Jagged1","FC"), type="ashr")

# summary(resLFC, alpha = 0.05)

plotMA(resLFC, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. FC - 72h\nAshr shrinked log2 fold changes")

# first remove the filtered genes (FDR=NA) and create a -log10(FDR) column
filtTab <- resLFC %>%
  as.data.frame() %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_72h) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol),
                   by = "ensembl") %>%
    mutate(`-log10(padj)` = -log10(padj)) %>%
  dplyr::filter(!is.na(`-log10(padj)`))

xlim <- range(filtTab$log2FoldChange)
xlim[1] <- ifelse(xlim[1] > -2, -2, xlim[1])
xlim[2] <- ifelse(xlim[2] < 2, 2, xlim[2])
xlim[1] <- ifelse(abs(xlim[2]) > abs(xlim[1]), -xlim[2], xlim[1])
xlim[2] <- ifelse(abs(xlim[1]) > abs(xlim[2]), -xlim[1], xlim[2])

ggplot(filtTab %>%
         dplyr::filter(!is.na(padj)), 
       aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol)) + 
  theme_classic() + 
  geom_point(aes(colour=padj < 0.05), size=1) +
  coord_cartesian(xlim = xlim) +
  geom_text_repel(data = filtTab %>%
                    dplyr::filter(padj < 0.05,
                                  abs(log2FoldChange) > 0.5),
                  aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol),
                  size = 3,
                  max.overlaps = 20,
                  force = 2) +
  scale_color_manual(values = as.character(merck_colors)[c(1,6)]) +
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed",
             color = "grey") +
  geom_vline(xintercept = 0,
             linetype = "dashed",
             color = "grey") +
  labs(x = "log2 fold Change",
       y = "-log10(p adj)",
       title = "Notch On (Jagged1) vs. Control (FC)")
  
```

### Jagged1 vs DAPT - 72h

```{r 11a-DE-Jagged1-DAPT-vulcano-72h, fig.width=4, fig.height=4, dev = "png"}

res_Jagged1_DAPT_72h <- results(ddsSE_72h, contrast=c("condition_treatment","Jagged1","DAPT"))

hist(res_Jagged1_DAPT_72h$pvalue)

#We can order our results table by the smallest p value:

res_Jagged1_DAPTOrdered <- res_Jagged1_DAPT_72h[order(res_Jagged1_DAPT_72h$padj),]
#We can summarize some basic tallies using the summary function.

summary(res_Jagged1_DAPT_72h, alpha = 0.05)

plotMA(res_Jagged1_DAPT_72h, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. DAPT - 72h\nlog2 fold changes")

printDF_Jagged1_DAPT <- res_Jagged1_DAPT_72h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_72h) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol, genename, entrez),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl) %>%
  dplyr::arrange(desc(log2FoldChange))

  DT::datatable(printDF_Jagged1_DAPT %>% 
          mutate_if(is.numeric, ~signif(., 2)), 
            class = 'cell-border stripe', 
            extensions = 'Buttons',
            rownames = TRUE,
            options = list(
              scrollX='600px',
              dom = 'Bfrtip',
              buttons = c('excel'),
              headerCallback = DT::JS(
                "function(thead) {",
                "  $(thead).css('font-size', '0.7em');",
                "}"
                )
              )
            ) %>%
    DT::formatStyle(columns = c(1:ncol(printDF_Jagged1_DAPT)), fontSize = '70%')
```

#### Vulcaono Plot - 72h

```{r 11b-DE-Jagged1-DAPT-vulcano-72h, fig.width=4, fig.height=4}

# resultsNames(ddsSE_72h)

resLFC <- lfcShrink(ddsSE_72h, contrast=c("condition_treatment","Jagged1","DAPT"), type="ashr")

# summary(resLFC, alpha = 0.05)

plotMA(resLFC, ylim=c(-2,2), cex=.2, alpha = 0.05)
abline(h=c(-0.5,0.5), col="dodgerblue", lwd=2)
title(main = "MA-plot Jagged1 vs. DAPT - 72h\nAshr shrinked log2 fold changes")

# first remove the filtered genes (FDR=NA) and create a -log10(FDR) column
filtTab <- resLFC %>%
  as.data.frame() %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_72h) %>%
                     as_tibble() %>%
                     dplyr::select(ensembl, symbol),
                   by = "ensembl") %>%
    mutate(`-log10(padj)` = -log10(padj)) %>%
  dplyr::filter(!is.na(`-log10(padj)`))

xlim <- range(filtTab$log2FoldChange)
xlim[1] <- ifelse(xlim[1] > -2, -2, xlim[1])
xlim[2] <- ifelse(xlim[2] < 2, 2, xlim[2])
xlim[1] <- ifelse(abs(xlim[2]) > abs(xlim[1]), -xlim[2], xlim[1])
xlim[2] <- ifelse(abs(xlim[1]) > abs(xlim[2]), -xlim[1], xlim[2])

ggplot(filtTab %>%
         dplyr::filter(!is.na(padj)), 
       aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol)) + 
  theme_classic() + 
  geom_point(aes(colour=padj < 0.05), size=1) +
  coord_cartesian(xlim = xlim) +
  geom_text_repel(data = filtTab %>%
                    dplyr::filter(padj < 0.05,
                                  abs(log2FoldChange) > 0.5),
                  aes(x = log2FoldChange, 
                    y=`-log10(padj)`,
                    label = symbol),
                  size = 3,
                  max.overlaps = 20,
                  force = 2) +
  scale_color_manual(values = as.character(merck_colors)[c(1,6)]) +
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed",
             color = "grey") +
  geom_vline(xintercept = 0,
             linetype = "dashed",
             color = "grey") +
  labs(x = "log2 fold Change",
       y = "-log10(p adj)",
       title = "Notch On (Jagged1) vs. Notch Off (DAPT)")
  
```

# Venn Diagrams

We load the results from the 8h timepoint and check, how many of the differentially regulated genes of the two timepoints overlap.  

## Jagged1 vs. DAPTs

### Upregulated in Jagged1 vs DAPT in 8h and 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

load("../RData/Part1b.RData")

printDF_Jagged1_DAPT_8h <- res_Jagged1_DAPT_8h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_preparation) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

printDF_Jagged1_DAPT_72h <- res_Jagged1_DAPT_72h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_preparation) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

overlap_Jagged_1_DAPT_up <- calculate.overlap(x = list(
    "Upregulated Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

print("Overlap Jagged1 vs. DAPT both upregulated:")
print(overlap_Jagged_1_DAPT_up$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in Jagged vs. DAPT treated cells\nin 8h and 72h treated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.01, 0.01),
  cat.pos = c(20, 40),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)
```

### Downregulated in Jagged1 vs DAPT in 8h and 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_DAPT_down <- calculate.overlap(x = list(
    "Downregulated Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

print("Overlap Jagged1 vs. DAPT both downregulated:")
print(overlap_Jagged_1_DAPT_down$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downpregulated Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in Jagged vs. DAPT treated cells\nin 8h and 72h treated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.01, 0.01),
  cat.pos = c(20, 40),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


### Downregulated in Jagged1 vs DAPT in 8h and Upregulated in Jagged1 vs DAPT in 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_DAPT_8hdown_72hup <- calculate.overlap(x = list(
    "Downregulated Genes Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Genes Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

print("Overlap Jagged1 vs. DAPT 8h downregulated and 72h upregulated:")
print(overlap_Jagged_1_DAPT_8hdown_72hup$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in 8h Jagged vs. DAPT treated cells\nand upregulated in 72h treated cells",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.005, 0.005),
  cat.pos = c(20, 120),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


### Upregulated in Jagged1 vs DAPT in 8h and Downregulated in Jagged1 vs DAPT in 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_DAPT_8hdown_72hup <- calculate.overlap(x = list(
    "Upregulated Genes Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Genes Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

# print("Overlap Jagged1 vs. DAPT 8h upregulated and 72h downregulated:")
# print(overlap_Jagged_1_DAPT_8hdown_72hup$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated Jagged1 vs. DAPT 8h" = printDF_Jagged1_DAPT_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Jagged1 vs. DAPT 72h" = printDF_Jagged1_DAPT_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in 8h Jagged vs. DAPT treated cells\nand Downregulated in 72h treated cells",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.005, 0.005),
  cat.pos = c(20, 120),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


## Jagged1 vs. FC

### Upregulated in Jagged1 vs DAPT in 8h and 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

printDF_Jagged1_FC_8h <- res_Jagged1_8h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_preparation) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

printDF_Jagged1_FC_72h <- res_Jagged1_72h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_preparation) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

overlap_Jagged_1_FC_up <- calculate.overlap(x = list(
    "Upregulated Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

print("Overlap Jagged1 vs. FC both upregulated:")
print(overlap_Jagged_1_FC_up$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in Jagged vs. FC treated cells\nin 8h and 72h treated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.01, 0.01),
  cat.pos = c(20, 40),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)
```

### Downregulated in Jagged1 vs FC in 8h and 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_FC_down <- calculate.overlap(x = list(
    "Downregulated Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

print("Overlap Jagged1 vs. FC both downregulated:")
print(overlap_Jagged_1_FC_down$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downpregulated Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in Jagged vs. FC treated cells\nin 8h and 72h treated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.01, 0.01),
  cat.pos = c(20, 40),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


### Downregulated in Jagged1 vs FC in 8h and Upregulated in Jagged1 vs FC in 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_FC_8hdown_72hup <- calculate.overlap(x = list(
    "Downregulated Genes Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Genes Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

# print("Overlap Jagged1 vs. FC 8h downregulated and 72h upregulated:")
# print(overlap_Jagged_1_FC_8hdown_72hup$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in 8h Jagged vs. FC treated cells\nand upregulated in 72h treated cells",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.005, 0.005),
  cat.pos = c(20, 120),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


### Upregulated in Jagged1 vs FC in 8h and Downregulated in Jagged1 vs FC in 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_FC_8hdown_72hup <- calculate.overlap(x = list(
    "Upregulated Genes Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Genes Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

# print("Overlap Jagged1 vs. FC 8h upregulated and 72h downregulated:")
# print(overlap_Jagged_1_FC_8hdown_72hup$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated Jagged1 vs. FC 8h" = printDF_Jagged1_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Jagged1 vs. FC 72h" = printDF_Jagged1_FC_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in 8h Jagged vs. FC treated cells\nand Downregulated in 72h treated cells",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.005, 0.005),
  cat.pos = c(20, 120),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


## DAPT vs. FC

### Upregulated in DAPT vs DAPT in 8h and 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

printDF_DAPT_FC_8h <- res_DAPT_8h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_preparation) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

printDF_DAPT_FC_72h <- res_DAPT_72h %>%
  as.data.frame() %>%
  dplyr::filter(!is.na(padj),
                padj < 0.05) %>%
  rownames_to_column("ensembl") %>%
  dplyr::mutate(ensembl = str_remove(ensembl, "\\.[0-9]+$")) %>%
  dplyr::left_join(rowData(ddsSE_preparation) %>%
                     as_tibble(),
                   by = c("ensembl")) %>%
  dplyr::select(-baseMean, -lfcSE, -stat, -ensembl, -gene_chrom) %>%
  dplyr::arrange(desc(log2FoldChange))

overlap_Jagged_1_FC_up <- calculate.overlap(x = list(
    "Upregulated DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

# print("Overlap DAPT vs. FC both upregulated:")
# print(overlap_Jagged_1_FC_up$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Upregulated DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in Jagged vs. FC treated cells\nin 8h and 72h treated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.01, 0.01),
  cat.pos = c(20, 40),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)
```

### Downregulated in DAPT vs FC in 8h and 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_FC_down <- calculate.overlap(x = list(
    "Downregulated DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Downregulated DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

print("Overlap DAPT vs. FC both downregulated:")
print(overlap_Jagged_1_FC_down$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downpregulated DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in Jagged vs. FC treated cells\nin 8h and 72h treated cell lines",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.01, 0.01),
  cat.pos = c(20, 40),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


### Downregulated in DAPT vs FC in 8h and Upregulated in DAPT vs FC in 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_FC_8hdown_72hup <- calculate.overlap(x = list(
    "Downregulated Genes DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated Genes DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)))

print("Overlap DAPT vs. FC 8h downregulated and 72h upregulated:")
print(overlap_Jagged_1_FC_8hdown_72hup$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Downregulated DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol),
    "Upregulated DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Downregulated Genes in 8h Jagged vs. FC treated cells\nand upregulated in 72h treated cells",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.005, 0.005),
  cat.pos = c(20, 120),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```


### Upregulated in DAPT vs FC in 8h and Downregulated in DAPT vs FC in 72h

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE, results='hide'}

overlap_Jagged_1_FC_8hdown_72hup <- calculate.overlap(x = list(
    "Upregulated Genes DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated Genes DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)))

# print("Overlap DAPT vs. FC 8h upregulated and 72h downregulated:")
# print(overlap_Jagged_1_FC_8hdown_72hup$a3)

venn.plot <- venn.diagram(
  disable.logging = TRUE,
  list(
    "Upregulated DAPT vs. FC 8h" = printDF_DAPT_FC_8h %>%
      dplyr::filter(log2FoldChange > 0) %>% 
      dplyr::pull(symbol),
    "Downregulated DAPT vs. FC 72h" = printDF_DAPT_FC_72h %>%
      dplyr::filter(log2FoldChange < 0) %>% 
      dplyr::pull(symbol)),
  filename = NULL,
  na = "remove",
  main = "Venn Diagram",
  main.fontfamily = "sans",
  main.fontface = "bold",
  sub = "Upregulated Genes in 8h Jagged vs. FC treated cells\nand Downregulated in 72h treated cells",
  sub.fontfamily = "sans",
  col = merck_colors[c(5,9)],
  fill = merck_colors[c(5,9)],
  alpha = 0.40,
  label.col = merck_colors[c(1)],
  fontfamily = "sans",
  fontface = "bold",
  cat.fontfamily = "sans",
  cat.dist = c(0.005, 0.005),
  cat.pos = c(20, 120),
  resolution = 300
)
grid.newpage()
grid.draw(venn.plot)

```

```{r 13-save-72h}

save(ddsSE_72h,
     res_Jagged1_DAPT_72h,
     res_Jagged1_72h,
     res_DAPT_72h,
     file = "../RData/Part1c.RData")
```

# Discussion of the Results and Recommendations

Differentially expression analysis of the training data set (72h) of the six basal-like cell lines resulted in lists of differentially expressed genes.  

Now, we will continue on XOP:  

Large RNASeq data set has been analyzed locally, now we want to include validation/optimization data sets.  

# References {.unlisted .unnumbered}

<div id="refs"></div>

# Appendix {.unlisted .unnumbered}
  
```{r SessionInfo}
pander::pander(sessionInfo())
```